// @react-spectrum/tag/dist/module.js
import "./main.css";
import {Item as $c14698d57a91452e$re_export$Item} from "@react-stately/collections";
import {ActionButton as $fUaQ2$ActionButton, ClearButton as $fUaQ2$ClearButton} from "@react-spectrum/button";
import {useTagGroup as $fUaQ2$useTagGroup, useTag as $fUaQ2$useTag} from "@react-aria/tag";
import {useDOMRef as $fUaQ2$useDOMRef, classNames as $fUaQ2$classNames, useStyleProps as $fUaQ2$useStyleProps, SlotProvider as $fUaQ2$SlotProvider, ClearSlots as $fUaQ2$ClearSlots} from "@react-spectrum/utils";
import {Field as $fUaQ2$Field} from "@react-spectrum/label";
import {FocusScope as $fUaQ2$FocusScope, FocusRing as $fUaQ2$FocusRing, useFocusRing as $fUaQ2$useFocusRing} from "@react-aria/focus";
import {useListState as $fUaQ2$useListState, ListCollection as $fUaQ2$ListCollection} from "@react-stately/list";
import {ListKeyboardDelegate as $fUaQ2$ListKeyboardDelegate} from "@react-aria/selection";
import {useProviderProps as $fUaQ2$useProviderProps, useProvider as $fUaQ2$useProvider, Provider as $fUaQ2$Provider} from "@react-spectrum/provider";
import $fUaQ2$react, {useRef as $fUaQ2$useRef, useState as $fUaQ2$useState, useMemo as $fUaQ2$useMemo, useCallback as $fUaQ2$useCallback, useEffect as $fUaQ2$useEffect} from "react";
import {useFormProps as $fUaQ2$useFormProps} from "@react-spectrum/form";
import {useValueEffect as $fUaQ2$useValueEffect, useId as $fUaQ2$useId, useResizeObserver as $fUaQ2$useResizeObserver, useLayoutEffect as $fUaQ2$useLayoutEffect, mergeProps as $fUaQ2$mergeProps} from "@react-aria/utils";
import {useLocale as $fUaQ2$useLocale, useLocalizedStringFormatter as $fUaQ2$useLocalizedStringFormatter} from "@react-aria/i18n";
import {Text as $fUaQ2$Text} from "@react-spectrum/text";
import {useHover as $fUaQ2$useHover} from "@react-aria/interactions";

function $parcel$interopDefault(a) {
  return a && a.__esModule ? a.default : a;
}
function $parcel$export(e, n, v, s) {
  Object.defineProperty(e, n, {get: v, set: s, enumerable: true, configurable: true});
}
/*
 * Copyright 2020 Adobe. All rights reserved.
 * This file is licensed to you under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License. You may obtain a copy
 * of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under
 * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 * OF ANY KIND, either express or implied. See the License for the specific language
 * governing permissions and limitations under the License.
 */ /*
 * Copyright 2020 Adobe. All rights reserved.
 * This file is licensed to you under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License. You may obtain a copy
 * of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under
 * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 * OF ANY KIND, either express or implied. See the License for the specific language
 * governing permissions and limitations under the License.
 */ 




var $fd5ee5726b29a200$exports = {};
var $4f97095b36611c9c$exports = {};
$4f97095b36611c9c$exports = {
    "actions": `الإجراءات`,
    "hideButtonLabel": `إظهار أقل`,
    "noTags": `بدون`,
    "showAllButtonLabel": (args, formatter)=>`عرض الكل (${formatter.number(args.tagCount)})`
};


var $d1606ced019b0a9e$exports = {};
$d1606ced019b0a9e$exports = {
    "actions": `Действия`,
    "hideButtonLabel": `Показване на по-малко`,
    "noTags": `Нито един`,
    "showAllButtonLabel": (args, formatter)=>`Показване на всички (${formatter.number(args.tagCount)})`
};


var $01e300f7d0078144$exports = {};
$01e300f7d0078144$exports = {
    "actions": `Akce`,
    "hideButtonLabel": `Zobrazit méně`,
    "noTags": `Žádný`,
    "showAllButtonLabel": (args, formatter)=>`Zobrazit vše (${formatter.number(args.tagCount)})`
};


var $f3862a5e5adf53a4$exports = {};
$f3862a5e5adf53a4$exports = {
    "actions": `Handlinger`,
    "hideButtonLabel": `Vis mindre`,
    "noTags": `Ingen`,
    "showAllButtonLabel": (args, formatter)=>`Vis alle (${formatter.number(args.tagCount)})`
};


var $74c0b6e11258b497$exports = {};
$74c0b6e11258b497$exports = {
    "actions": `Aktionen`,
    "hideButtonLabel": `Weniger zeigen`,
    "noTags": `Keine`,
    "showAllButtonLabel": (args, formatter)=>`Alle anzeigen (${formatter.number(args.tagCount)})`
};


var $54fc4e3090259431$exports = {};
$54fc4e3090259431$exports = {
    "actions": `Ενέργειες`,
    "hideButtonLabel": `Εμφάνιση λιγότερων`,
    "noTags": `Κανένα`,
    "showAllButtonLabel": (args, formatter)=>`Εμφάνιση όλων (${formatter.number(args.tagCount)})`
};


var $edf02907a7caaf96$exports = {};
$edf02907a7caaf96$exports = {
    "showAllButtonLabel": (args, formatter)=>`Show all (${formatter.number(args.tagCount)})`,
    "hideButtonLabel": `Show less`,
    "actions": `Actions`,
    "noTags": `None`
};


var $b15ad3cf4b213b8e$exports = {};
$b15ad3cf4b213b8e$exports = {
    "actions": `Acciones`,
    "hideButtonLabel": `Mostrar menos`,
    "noTags": `Ninguno`,
    "showAllButtonLabel": (args, formatter)=>`Mostrar todo (${formatter.number(args.tagCount)})`
};


var $36cfda096c3db653$exports = {};
$36cfda096c3db653$exports = {
    "actions": `Toimingud`,
    "hideButtonLabel": `Kuva vähem`,
    "noTags": `Puudub`,
    "showAllButtonLabel": (args, formatter)=>`Kuva kõik (${formatter.number(args.tagCount)})`
};


var $1a42e74299cab64d$exports = {};
$1a42e74299cab64d$exports = {
    "actions": `Toiminnot`,
    "hideButtonLabel": `Näytä vähemmän`,
    "noTags": `Ei mitään`,
    "showAllButtonLabel": (args, formatter)=>`Näytä kaikki (${formatter.number(args.tagCount)})`
};


var $dd8d3f41a5bfc373$exports = {};
$dd8d3f41a5bfc373$exports = {
    "actions": `Actions`,
    "hideButtonLabel": `Afficher moins`,
    "noTags": `Aucun`,
    "showAllButtonLabel": (args, formatter)=>`Tout afficher (${formatter.number(args.tagCount)})`
};


var $35d68da324c3ed3e$exports = {};
$35d68da324c3ed3e$exports = {
    "actions": `פעולות`,
    "hideButtonLabel": `הצג פחות`,
    "noTags": `ללא`,
    "showAllButtonLabel": (args, formatter)=>`הצג הכל (${formatter.number(args.tagCount)})`
};


var $92746ce1284cd29f$exports = {};
$92746ce1284cd29f$exports = {
    "actions": `Radnje`,
    "hideButtonLabel": `Prikaži manje`,
    "noTags": `Nema`,
    "showAllButtonLabel": (args, formatter)=>`Prikaži sve (${formatter.number(args.tagCount)})`
};


var $570956f4f59f34b9$exports = {};
$570956f4f59f34b9$exports = {
    "actions": `Műveletek`,
    "hideButtonLabel": `Mutass kevesebbet`,
    "noTags": `Egyik sem`,
    "showAllButtonLabel": (args, formatter)=>`Az összes megjelenítése (${formatter.number(args.tagCount)})`
};


var $f680d16d6f0d92d8$exports = {};
$f680d16d6f0d92d8$exports = {
    "actions": `Azioni`,
    "hideButtonLabel": `Mostra meno`,
    "noTags": `Nessuno`,
    "showAllButtonLabel": (args, formatter)=>`Mostra tutto (${formatter.number(args.tagCount)})`
};


var $d28c56f33dc08452$exports = {};
$d28c56f33dc08452$exports = {
    "actions": `アクション`,
    "hideButtonLabel": `表示を減らす`,
    "noTags": `なし`,
    "showAllButtonLabel": (args, formatter)=>`すべての (${formatter.number(args.tagCount)}) を表示`
};


var $fc6f973338bd2d9c$exports = {};
$fc6f973338bd2d9c$exports = {
    "actions": `액션`,
    "hideButtonLabel": `간단히 표시`,
    "noTags": `없음`,
    "showAllButtonLabel": (args, formatter)=>`모두 표시 (${formatter.number(args.tagCount)})`
};


var $8664f3f6a18cfaa6$exports = {};
$8664f3f6a18cfaa6$exports = {
    "actions": `Veiksmai`,
    "hideButtonLabel": `Rodyti mažiau`,
    "noTags": `Nėra`,
    "showAllButtonLabel": (args, formatter)=>`Rodyti viską (${formatter.number(args.tagCount)})`
};


var $bd6e0fb18b482534$exports = {};
$bd6e0fb18b482534$exports = {
    "actions": `Darbības`,
    "hideButtonLabel": `Rādīt mazāk`,
    "noTags": `Nav`,
    "showAllButtonLabel": (args, formatter)=>`Rādīt visu (${formatter.number(args.tagCount)})`
};


var $3df66303c61fb25b$exports = {};
$3df66303c61fb25b$exports = {
    "actions": `Handlinger`,
    "hideButtonLabel": `Vis mindre`,
    "noTags": `Ingen`,
    "showAllButtonLabel": (args, formatter)=>`Vis alle (${formatter.number(args.tagCount)})`
};


var $0f533b61cdf70176$exports = {};
$0f533b61cdf70176$exports = {
    "actions": `Acties`,
    "hideButtonLabel": `Minder weergeven`,
    "noTags": `Geen`,
    "showAllButtonLabel": (args, formatter)=>`Alles tonen (${formatter.number(args.tagCount)})`
};


var $03a6cd8f61b18787$exports = {};
$03a6cd8f61b18787$exports = {
    "actions": `Działania`,
    "hideButtonLabel": `Wyświetl mniej`,
    "noTags": `Brak`,
    "showAllButtonLabel": (args, formatter)=>`Pokaż wszystko (${formatter.number(args.tagCount)})`
};


var $c726a9d931f60c66$exports = {};
$c726a9d931f60c66$exports = {
    "actions": `Ações`,
    "hideButtonLabel": `Mostrar menos`,
    "noTags": `Nenhum`,
    "showAllButtonLabel": (args, formatter)=>`Mostrar tudo (${formatter.number(args.tagCount)})`
};


var $ae9982f17f7235eb$exports = {};
$ae9982f17f7235eb$exports = {
    "actions": `Ações`,
    "hideButtonLabel": `Mostrar menos`,
    "noTags": `Nenhum`,
    "showAllButtonLabel": (args, formatter)=>`Mostrar tudo (${formatter.number(args.tagCount)})`
};


var $c756723b814c63fa$exports = {};
$c756723b814c63fa$exports = {
    "actions": `Acțiuni`,
    "hideButtonLabel": `Se afișează mai puțin`,
    "noTags": `Niciuna`,
    "showAllButtonLabel": (args, formatter)=>`Se afișează tot (${formatter.number(args.tagCount)})`
};


var $1601284ef6ec0730$exports = {};
$1601284ef6ec0730$exports = {
    "actions": `Действия`,
    "hideButtonLabel": `Показать меньше`,
    "noTags": `Нет`,
    "showAllButtonLabel": (args, formatter)=>`Показать все (${formatter.number(args.tagCount)})`
};


var $0388aac8e3bc5b59$exports = {};
$0388aac8e3bc5b59$exports = {
    "actions": `Akcie`,
    "hideButtonLabel": `Zobraziť menej`,
    "noTags": `Žiadne`,
    "showAllButtonLabel": (args, formatter)=>`Zobraziť všetko (${formatter.number(args.tagCount)})`
};


var $247d9c4a71da6b08$exports = {};
$247d9c4a71da6b08$exports = {
    "actions": `Dejanja`,
    "hideButtonLabel": `Prikaži manj`,
    "noTags": `Nič`,
    "showAllButtonLabel": (args, formatter)=>`Prikaž vse (${formatter.number(args.tagCount)})`
};


var $91424dbafdc02f68$exports = {};
$91424dbafdc02f68$exports = {
    "actions": `Radnje`,
    "hideButtonLabel": `Prikaži manje`,
    "noTags": `Ne postoji`,
    "showAllButtonLabel": (args, formatter)=>`Prikaži sve (${formatter.number(args.tagCount)})`
};


var $23c23096b6c1b0fa$exports = {};
$23c23096b6c1b0fa$exports = {
    "actions": `Åtgärder`,
    "hideButtonLabel": `Visa mindre`,
    "noTags": `Ingen`,
    "showAllButtonLabel": (args, formatter)=>`Visa alla (${formatter.number(args.tagCount)})`
};


var $d82875ae28963e58$exports = {};
$d82875ae28963e58$exports = {
    "actions": `Eylemler`,
    "hideButtonLabel": `Daha az göster`,
    "noTags": `Hiçbiri`,
    "showAllButtonLabel": (args, formatter)=>`Tümünü göster (${formatter.number(args.tagCount)})`
};


var $1dd153688f5574fc$exports = {};
$1dd153688f5574fc$exports = {
    "actions": `Дії`,
    "hideButtonLabel": `Показувати менше`,
    "noTags": `Немає`,
    "showAllButtonLabel": (args, formatter)=>`Показати всі (${formatter.number(args.tagCount)})`
};


var $6427307c9596258c$exports = {};
$6427307c9596258c$exports = {
    "actions": `操作`,
    "hideButtonLabel": `显示更少`,
    "noTags": `无`,
    "showAllButtonLabel": (args, formatter)=>`全部显示 (${formatter.number(args.tagCount)})`
};


var $18b85da2aafaaf3e$exports = {};
$18b85da2aafaaf3e$exports = {
    "actions": `動作`,
    "hideButtonLabel": `顯示較少`,
    "noTags": `無`,
    "showAllButtonLabel": (args, formatter)=>`顯示全部 (${formatter.number(args.tagCount)})`
};


$fd5ee5726b29a200$exports = {
    "ar-AE": $4f97095b36611c9c$exports,
    "bg-BG": $d1606ced019b0a9e$exports,
    "cs-CZ": $01e300f7d0078144$exports,
    "da-DK": $f3862a5e5adf53a4$exports,
    "de-DE": $74c0b6e11258b497$exports,
    "el-GR": $54fc4e3090259431$exports,
    "en-US": $edf02907a7caaf96$exports,
    "es-ES": $b15ad3cf4b213b8e$exports,
    "et-EE": $36cfda096c3db653$exports,
    "fi-FI": $1a42e74299cab64d$exports,
    "fr-FR": $dd8d3f41a5bfc373$exports,
    "he-IL": $35d68da324c3ed3e$exports,
    "hr-HR": $92746ce1284cd29f$exports,
    "hu-HU": $570956f4f59f34b9$exports,
    "it-IT": $f680d16d6f0d92d8$exports,
    "ja-JP": $d28c56f33dc08452$exports,
    "ko-KR": $fc6f973338bd2d9c$exports,
    "lt-LT": $8664f3f6a18cfaa6$exports,
    "lv-LV": $bd6e0fb18b482534$exports,
    "nb-NO": $3df66303c61fb25b$exports,
    "nl-NL": $0f533b61cdf70176$exports,
    "pl-PL": $03a6cd8f61b18787$exports,
    "pt-BR": $c726a9d931f60c66$exports,
    "pt-PT": $ae9982f17f7235eb$exports,
    "ro-RO": $c756723b814c63fa$exports,
    "ru-RU": $1601284ef6ec0730$exports,
    "sk-SK": $0388aac8e3bc5b59$exports,
    "sl-SI": $247d9c4a71da6b08$exports,
    "sr-SP": $91424dbafdc02f68$exports,
    "sv-SE": $23c23096b6c1b0fa$exports,
    "tr-TR": $d82875ae28963e58$exports,
    "uk-UA": $1dd153688f5574fc$exports,
    "zh-CN": $6427307c9596258c$exports,
    "zh-TW": $18b85da2aafaaf3e$exports
};






var $c033a7c057e84ef1$exports = {};

$parcel$export($c033a7c057e84ef1$exports, "spectrum-Tag", () => $c033a7c057e84ef1$export$6174bd1d20b4c022, (v) => $c033a7c057e84ef1$export$6174bd1d20b4c022 = v);
$parcel$export($c033a7c057e84ef1$exports, "focus-ring", () => $c033a7c057e84ef1$export$f39a09f249340e2a, (v) => $c033a7c057e84ef1$export$f39a09f249340e2a = v);
$parcel$export($c033a7c057e84ef1$exports, "spectrum-Tags", () => $c033a7c057e84ef1$export$eb7e80013a1212da, (v) => $c033a7c057e84ef1$export$eb7e80013a1212da = v);
$parcel$export($c033a7c057e84ef1$exports, "spectrum-Tags-container", () => $c033a7c057e84ef1$export$3300169c5d27fe96, (v) => $c033a7c057e84ef1$export$3300169c5d27fe96 = v);
$parcel$export($c033a7c057e84ef1$exports, "spectrum-Tags-container--empty", () => $c033a7c057e84ef1$export$a5e59d283417e02b, (v) => $c033a7c057e84ef1$export$a5e59d283417e02b = v);
$parcel$export($c033a7c057e84ef1$exports, "is-disabled", () => $c033a7c057e84ef1$export$d35bc1e505d1ebbf, (v) => $c033a7c057e84ef1$export$d35bc1e505d1ebbf = v);
$parcel$export($c033a7c057e84ef1$exports, "spectrum-Tag-removeButton", () => $c033a7c057e84ef1$export$23a76b996f0a6c1, (v) => $c033a7c057e84ef1$export$23a76b996f0a6c1 = v);
$parcel$export($c033a7c057e84ef1$exports, "spectrum-Tag-cell", () => $c033a7c057e84ef1$export$9c4fd4d7faada408, (v) => $c033a7c057e84ef1$export$9c4fd4d7faada408 = v);
$parcel$export($c033a7c057e84ef1$exports, "spectrum-Tag-icon", () => $c033a7c057e84ef1$export$21f5254139e14871, (v) => $c033a7c057e84ef1$export$21f5254139e14871 = v);
$parcel$export($c033a7c057e84ef1$exports, "spectrum-Tag-avatar", () => $c033a7c057e84ef1$export$7b57f5221a1b5306, (v) => $c033a7c057e84ef1$export$7b57f5221a1b5306 = v);
$parcel$export($c033a7c057e84ef1$exports, "spectrum-Tag-content", () => $c033a7c057e84ef1$export$d281c967f4ee071e, (v) => $c033a7c057e84ef1$export$d281c967f4ee071e = v);
$parcel$export($c033a7c057e84ef1$exports, "spectrum-Tag--removable", () => $c033a7c057e84ef1$export$87165794583be111, (v) => $c033a7c057e84ef1$export$87165794583be111 = v);
$parcel$export($c033a7c057e84ef1$exports, "spectrum-Tags-actions", () => $c033a7c057e84ef1$export$8afaa02110915bbb, (v) => $c033a7c057e84ef1$export$8afaa02110915bbb = v);
$parcel$export($c033a7c057e84ef1$exports, "spectrum-Tags-actionButton", () => $c033a7c057e84ef1$export$e6882f8db018e3bd, (v) => $c033a7c057e84ef1$export$e6882f8db018e3bd = v);
$parcel$export($c033a7c057e84ef1$exports, "spectrum-Tags-fieldWrapper", () => $c033a7c057e84ef1$export$c9b7ca327d69b2b5, (v) => $c033a7c057e84ef1$export$c9b7ca327d69b2b5 = v);
$parcel$export($c033a7c057e84ef1$exports, "spectrum-Tags-fieldWrapper--positionSide", () => $c033a7c057e84ef1$export$a7c11d4acc244e09, (v) => $c033a7c057e84ef1$export$a7c11d4acc244e09 = v);
$parcel$export($c033a7c057e84ef1$exports, "spectrum-Tags-empty-state", () => $c033a7c057e84ef1$export$cd48f925394bb16, (v) => $c033a7c057e84ef1$export$cd48f925394bb16 = v);
$parcel$export($c033a7c057e84ef1$exports, "is-hovered", () => $c033a7c057e84ef1$export$b8813cd5d7824ce7, (v) => $c033a7c057e84ef1$export$b8813cd5d7824ce7 = v);
$parcel$export($c033a7c057e84ef1$exports, "is-focused", () => $c033a7c057e84ef1$export$e7dc768d35940237, (v) => $c033a7c057e84ef1$export$e7dc768d35940237 = v);
var $c033a7c057e84ef1$export$6174bd1d20b4c022;
var $c033a7c057e84ef1$export$f39a09f249340e2a;
var $c033a7c057e84ef1$export$eb7e80013a1212da;
var $c033a7c057e84ef1$export$3300169c5d27fe96;
var $c033a7c057e84ef1$export$a5e59d283417e02b;
var $c033a7c057e84ef1$export$d35bc1e505d1ebbf;
var $c033a7c057e84ef1$export$23a76b996f0a6c1;
var $c033a7c057e84ef1$export$9c4fd4d7faada408;
var $c033a7c057e84ef1$export$21f5254139e14871;
var $c033a7c057e84ef1$export$7b57f5221a1b5306;
var $c033a7c057e84ef1$export$d281c967f4ee071e;
var $c033a7c057e84ef1$export$87165794583be111;
var $c033a7c057e84ef1$export$8afaa02110915bbb;
var $c033a7c057e84ef1$export$e6882f8db018e3bd;
var $c033a7c057e84ef1$export$c9b7ca327d69b2b5;
var $c033a7c057e84ef1$export$a7c11d4acc244e09;
var $c033a7c057e84ef1$export$cd48f925394bb16;
var $c033a7c057e84ef1$export$b8813cd5d7824ce7;
var $c033a7c057e84ef1$export$e7dc768d35940237;
$c033a7c057e84ef1$export$6174bd1d20b4c022 = "spectrum-Tag_8407eb";
$c033a7c057e84ef1$export$f39a09f249340e2a = "focus-ring_8407eb";
$c033a7c057e84ef1$export$eb7e80013a1212da = "spectrum-Tags_8407eb";
$c033a7c057e84ef1$export$3300169c5d27fe96 = "spectrum-Tags-container_8407eb";
$c033a7c057e84ef1$export$a5e59d283417e02b = "spectrum-Tags-container--empty_8407eb";
$c033a7c057e84ef1$export$d35bc1e505d1ebbf = "is-disabled_8407eb";
$c033a7c057e84ef1$export$23a76b996f0a6c1 = "spectrum-Tag-removeButton_8407eb";
$c033a7c057e84ef1$export$9c4fd4d7faada408 = "spectrum-Tag-cell_8407eb";
$c033a7c057e84ef1$export$21f5254139e14871 = "spectrum-Tag-icon_8407eb";
$c033a7c057e84ef1$export$7b57f5221a1b5306 = "spectrum-Tag-avatar_8407eb";
$c033a7c057e84ef1$export$d281c967f4ee071e = "spectrum-Tag-content_8407eb";
$c033a7c057e84ef1$export$87165794583be111 = "spectrum-Tag--removable_8407eb";
$c033a7c057e84ef1$export$8afaa02110915bbb = "spectrum-Tags-actions_8407eb";
$c033a7c057e84ef1$export$e6882f8db018e3bd = "spectrum-Tags-actionButton_8407eb";
$c033a7c057e84ef1$export$c9b7ca327d69b2b5 = "spectrum-Tags-fieldWrapper_8407eb";
$c033a7c057e84ef1$export$a7c11d4acc244e09 = "spectrum-Tags-fieldWrapper--positionSide_8407eb";
$c033a7c057e84ef1$export$cd48f925394bb16 = "spectrum-Tags-empty-state_8407eb";
$c033a7c057e84ef1$export$b8813cd5d7824ce7 = "is-hovered_8407eb";
$c033a7c057e84ef1$export$e7dc768d35940237 = "is-focused_8407eb";


/*
 * Copyright 2020 Adobe. All rights reserved.
 * This file is licensed to you under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License. You may obtain a copy
 * of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under
 * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 * OF ANY KIND, either express or implied. See the License for the specific language
 * governing permissions and limitations under the License.
 */ 








function $b5f5ee35d3b6b95c$export$3288d34c523a1192(props) {
    const { item: item , state: state , ...otherProps } = props;
    // @ts-ignore
    let { styleProps: styleProps  } = (0, $fUaQ2$useStyleProps)(otherProps);
    let { hoverProps: hoverProps , isHovered: isHovered  } = (0, $fUaQ2$useHover)({});
    let { isFocused: isFocused , isFocusVisible: isFocusVisible , focusProps: focusProps  } = (0, $fUaQ2$useFocusRing)({
        within: true
    });
    let ref = (0, $fUaQ2$useRef)();
    let { removeButtonProps: removeButtonProps , gridCellProps: gridCellProps , rowProps: rowProps , allowsRemoving: allowsRemoving  } = (0, $fUaQ2$useTag)({
        ...props,
        item: item
    }, state, ref);
    return /*#__PURE__*/ (0, $fUaQ2$react).createElement("div", {
        ...(0, $fUaQ2$mergeProps)(rowProps, hoverProps, focusProps),
        className: (0, $fUaQ2$classNames)((0, (/*@__PURE__*/$parcel$interopDefault($c033a7c057e84ef1$exports))), "spectrum-Tag", {
            "focus-ring": isFocusVisible,
            "is-focused": isFocused,
            "is-hovered": isHovered,
            "spectrum-Tag--removable": allowsRemoving
        }, styleProps.className),
        ref: ref
    }, /*#__PURE__*/ (0, $fUaQ2$react).createElement("div", {
        className: (0, $fUaQ2$classNames)((0, (/*@__PURE__*/$parcel$interopDefault($c033a7c057e84ef1$exports))), "spectrum-Tag-cell"),
        ...gridCellProps
    }, /*#__PURE__*/ (0, $fUaQ2$react).createElement((0, $fUaQ2$SlotProvider), {
        slots: {
            icon: {
                UNSAFE_className: (0, $fUaQ2$classNames)((0, (/*@__PURE__*/$parcel$interopDefault($c033a7c057e84ef1$exports))), "spectrum-Tag-icon"),
                size: "XS"
            },
            text: {
                UNSAFE_className: (0, $fUaQ2$classNames)((0, (/*@__PURE__*/$parcel$interopDefault($c033a7c057e84ef1$exports))), "spectrum-Tag-content")
            },
            avatar: {
                UNSAFE_className: (0, $fUaQ2$classNames)((0, (/*@__PURE__*/$parcel$interopDefault($c033a7c057e84ef1$exports))), "spectrum-Tag-avatar"),
                size: "avatar-size-50"
            }
        }
    }, typeof item.rendered === "string" ? /*#__PURE__*/ (0, $fUaQ2$react).createElement((0, $fUaQ2$Text), null, item.rendered) : item.rendered, /*#__PURE__*/ (0, $fUaQ2$react).createElement((0, $fUaQ2$ClearSlots), null, allowsRemoving && /*#__PURE__*/ (0, $fUaQ2$react).createElement($b5f5ee35d3b6b95c$var$TagRemoveButton, {
        item: item,
        ...removeButtonProps,
        UNSAFE_className: (0, $fUaQ2$classNames)((0, (/*@__PURE__*/$parcel$interopDefault($c033a7c057e84ef1$exports))), "spectrum-Tag-removeButton")
    })))));
}
function $b5f5ee35d3b6b95c$var$TagRemoveButton(props) {
    let { styleProps: styleProps  } = (0, $fUaQ2$useStyleProps)(props);
    return /*#__PURE__*/ (0, $fUaQ2$react).createElement("span", styleProps, /*#__PURE__*/ (0, $fUaQ2$react).createElement((0, $fUaQ2$ClearButton), props));
}





const $880d112c7a588209$var$TAG_STYLES = {
    medium: {
        height: 24,
        margin: 4
    },
    large: {
        height: 30,
        margin: 5
    }
};
function $880d112c7a588209$var$TagGroup(props, ref) {
    props = (0, $fUaQ2$useProviderProps)(props);
    props = (0, $fUaQ2$useFormProps)(props);
    let { maxRows: maxRows , children: children , actionLabel: actionLabel , onAction: onAction , labelPosition: labelPosition , renderEmptyState: renderEmptyState = ()=>stringFormatter.format("noTags")  } = props;
    let domRef = (0, $fUaQ2$useDOMRef)(ref);
    let containerRef = (0, $fUaQ2$useRef)(null);
    let tagsRef = (0, $fUaQ2$useRef)(null);
    let { direction: direction  } = (0, $fUaQ2$useLocale)();
    let { scale: scale  } = (0, $fUaQ2$useProvider)();
    let stringFormatter = (0, $fUaQ2$useLocalizedStringFormatter)((0, (/*@__PURE__*/$parcel$interopDefault($fd5ee5726b29a200$exports))));
    let [isCollapsed, setIsCollapsed] = (0, $fUaQ2$useState)(maxRows != null);
    let state = (0, $fUaQ2$useListState)(props);
    let [tagState, setTagState] = (0, $fUaQ2$useValueEffect)({
        visibleTagCount: state.collection.size,
        showCollapseButton: false
    });
    let keyboardDelegate = (0, $fUaQ2$useMemo)(()=>{
        let collection = isCollapsed ? new (0, $fUaQ2$ListCollection)([
            ...state.collection
        ].slice(0, tagState.visibleTagCount)) : new (0, $fUaQ2$ListCollection)([
            ...state.collection
        ]);
        return new (0, $fUaQ2$ListKeyboardDelegate)({
            collection: collection,
            ref: domRef,
            direction: direction,
            orientation: "horizontal"
        });
    }, [
        direction,
        isCollapsed,
        state.collection,
        tagState.visibleTagCount,
        domRef
    ]);
    // Remove onAction from props so it doesn't make it into useGridList.
    delete props.onAction;
    let { gridProps: gridProps , labelProps: labelProps , descriptionProps: descriptionProps , errorMessageProps: errorMessageProps  } = (0, $fUaQ2$useTagGroup)({
        ...props,
        keyboardDelegate: keyboardDelegate
    }, state, tagsRef);
    let actionsId = (0, $fUaQ2$useId)();
    let actionsRef = (0, $fUaQ2$useRef)(null);
    let updateVisibleTagCount = (0, $fUaQ2$useCallback)(()=>{
        if (maxRows > 0) {
            let computeVisibleTagCount = ()=>{
                // Refs can be null at runtime.
                let currContainerRef = containerRef.current;
                let currTagsRef = tagsRef.current;
                let currActionsRef = actionsRef.current;
                if (!currContainerRef || !currTagsRef || !currActionsRef || state.collection.size === 0) return {
                    visibleTagCount: 0,
                    showCollapseButton: false
                };
                // Count rows and show tags until we hit the maxRows.
                let tags = [
                    ...currTagsRef.children
                ];
                let currY = -Infinity;
                let rowCount = 0;
                let index = 0;
                let tagWidths = [];
                for (let tag of tags){
                    let { width: width , y: y  } = tag.getBoundingClientRect();
                    if (y !== currY) {
                        currY = y;
                        rowCount++;
                    }
                    if (rowCount > maxRows) break;
                    tagWidths.push(width);
                    index++;
                }
                // Remove tags until there is space for the collapse button and action button (if present) on the last row.
                let buttons = [
                    ...currActionsRef.children
                ];
                if (buttons.length > 0 && rowCount >= maxRows) {
                    var _tags_;
                    let buttonsWidth = buttons.reduce((acc, curr)=>acc += curr.getBoundingClientRect().width, 0);
                    buttonsWidth += $880d112c7a588209$var$TAG_STYLES[scale].margin * 2 * buttons.length;
                    let end = direction === "ltr" ? "right" : "left";
                    let containerEnd = currContainerRef.parentElement.getBoundingClientRect()[end];
                    let lastTagEnd = (_tags_ = tags[index - 1]) === null || _tags_ === void 0 ? void 0 : _tags_.getBoundingClientRect()[end];
                    lastTagEnd += $880d112c7a588209$var$TAG_STYLES[scale].margin;
                    let availableWidth = containerEnd - lastTagEnd;
                    while(availableWidth < buttonsWidth && index > 0){
                        availableWidth += tagWidths.pop();
                        index--;
                    }
                }
                return {
                    visibleTagCount: Math.max(index, 1),
                    showCollapseButton: index < state.collection.size
                };
            };
            setTagState(function*() {
                // Update to show all items.
                yield {
                    visibleTagCount: state.collection.size,
                    showCollapseButton: true
                };
                // Measure, and update to show the items until maxRows is reached.
                yield computeVisibleTagCount();
            });
        }
    }, [
        maxRows,
        setTagState,
        direction,
        scale,
        state.collection.size
    ]);
    (0, $fUaQ2$useResizeObserver)({
        ref: containerRef,
        onResize: updateVisibleTagCount
    });
    // eslint-disable-next-line react-hooks/exhaustive-deps
    (0, $fUaQ2$useLayoutEffect)(updateVisibleTagCount, [
        children
    ]);
    (0, $fUaQ2$useEffect)(()=>{
        var // Recalculate visible tags when fonts are loaded.
        _document_fonts;
        (_document_fonts = document.fonts) === null || _document_fonts === void 0 ? void 0 : _document_fonts.ready.then(()=>updateVisibleTagCount());
    // eslint-disable-next-line react-hooks/exhaustive-deps
    }, []);
    let visibleTags = (0, $fUaQ2$useMemo)(()=>[
            ...state.collection
        ].slice(0, isCollapsed ? tagState.visibleTagCount : state.collection.size), [
        isCollapsed,
        state.collection,
        tagState.visibleTagCount
    ]);
    let handlePressCollapse = ()=>{
        // Prevents button from losing focus if focusedKey got collapsed.
        state.selectionManager.setFocusedKey(null);
        setIsCollapsed((prevCollapsed)=>!prevCollapsed);
    };
    let showActions = tagState.showCollapseButton || actionLabel && onAction;
    let isEmpty = state.collection.size === 0;
    let containerStyle = (0, $fUaQ2$useMemo)(()=>{
        if (maxRows == null || !isCollapsed || isEmpty) return undefined;
        let maxHeight = ($880d112c7a588209$var$TAG_STYLES[scale].height + $880d112c7a588209$var$TAG_STYLES[scale].margin * 2) * maxRows;
        return {
            maxHeight: maxHeight,
            overflow: "hidden"
        };
    }, [
        isCollapsed,
        maxRows,
        isEmpty,
        scale
    ]);
    return /*#__PURE__*/ (0, $fUaQ2$react).createElement((0, $fUaQ2$FocusScope), null, /*#__PURE__*/ (0, $fUaQ2$react).createElement((0, $fUaQ2$Field), {
        ...props,
        labelProps: labelProps,
        descriptionProps: descriptionProps,
        errorMessageProps: errorMessageProps,
        showErrorIcon: true,
        ref: domRef,
        elementType: "span",
        wrapperClassName: (0, $fUaQ2$classNames)((0, (/*@__PURE__*/$parcel$interopDefault($c033a7c057e84ef1$exports))), "spectrum-Tags-fieldWrapper", {
            "spectrum-Tags-fieldWrapper--positionSide": labelPosition === "side"
        })
    }, /*#__PURE__*/ (0, $fUaQ2$react).createElement("div", {
        ref: containerRef,
        style: containerStyle,
        className: (0, $fUaQ2$classNames)((0, (/*@__PURE__*/$parcel$interopDefault($c033a7c057e84ef1$exports))), "spectrum-Tags-container", {
            "spectrum-Tags-container--empty": isEmpty
        })
    }, /*#__PURE__*/ (0, $fUaQ2$react).createElement((0, $fUaQ2$FocusRing), {
        focusRingClass: (0, $fUaQ2$classNames)((0, (/*@__PURE__*/$parcel$interopDefault($c033a7c057e84ef1$exports))), "focus-ring")
    }, /*#__PURE__*/ (0, $fUaQ2$react).createElement("div", {
        ref: tagsRef,
        ...gridProps,
        className: (0, $fUaQ2$classNames)((0, (/*@__PURE__*/$parcel$interopDefault($c033a7c057e84ef1$exports))), "spectrum-Tags")
    }, visibleTags.map((item)=>/*#__PURE__*/ (0, $fUaQ2$react).createElement((0, $b5f5ee35d3b6b95c$export$3288d34c523a1192), {
            ...item.props,
            key: item.key,
            item: item,
            state: state
        }, item.rendered)), isEmpty && /*#__PURE__*/ (0, $fUaQ2$react).createElement("div", {
        className: (0, $fUaQ2$classNames)((0, (/*@__PURE__*/$parcel$interopDefault($c033a7c057e84ef1$exports))), "spectrum-Tags-empty-state")
    }, renderEmptyState()))), showActions && !isEmpty && /*#__PURE__*/ (0, $fUaQ2$react).createElement((0, $fUaQ2$Provider), {
        isDisabled: false
    }, /*#__PURE__*/ (0, $fUaQ2$react).createElement("div", {
        role: "group",
        ref: actionsRef,
        id: actionsId,
        "aria-label": stringFormatter.format("actions"),
        "aria-labelledby": `${gridProps.id} ${actionsId}`,
        className: (0, $fUaQ2$classNames)((0, (/*@__PURE__*/$parcel$interopDefault($c033a7c057e84ef1$exports))), "spectrum-Tags-actions")
    }, tagState.showCollapseButton && /*#__PURE__*/ (0, $fUaQ2$react).createElement((0, $fUaQ2$ActionButton), {
        isQuiet: true,
        onPress: handlePressCollapse,
        UNSAFE_className: (0, $fUaQ2$classNames)((0, (/*@__PURE__*/$parcel$interopDefault($c033a7c057e84ef1$exports))), "spectrum-Tags-actionButton")
    }, isCollapsed ? stringFormatter.format("showAllButtonLabel", {
        tagCount: state.collection.size
    }) : stringFormatter.format("hideButtonLabel")), actionLabel && onAction && /*#__PURE__*/ (0, $fUaQ2$react).createElement((0, $fUaQ2$ActionButton), {
        isQuiet: true,
        onPress: ()=>{
            return onAction === null || onAction === void 0 ? void 0 : onAction();
        },
        UNSAFE_className: (0, $fUaQ2$classNames)((0, (/*@__PURE__*/$parcel$interopDefault($c033a7c057e84ef1$exports))), "spectrum-Tags-actionButton")
    }, actionLabel))))));
}
/** Tags allow users to categorize content. They can represent keywords or people, and are grouped to describe an item or a search request. */ const $880d112c7a588209$export$67ea30858aaf75e3 = /*#__PURE__*/ (0, $fUaQ2$react).forwardRef($880d112c7a588209$var$TagGroup);





export {$880d112c7a588209$export$67ea30858aaf75e3 as TagGroup, $c14698d57a91452e$re_export$Item as Item};
