// @react-spectrum/actionbar/dist/import.mjs
import "./main.css";
import {Item as $da70baa779e17e7a$re_export$Item} from "@react-stately/collections";
import {ActionButton as $N9qJl$ActionButton} from "@react-spectrum/button";
import {ActionGroup as $N9qJl$ActionGroup} from "@react-spectrum/actiongroup";
import {announce as $N9qJl$announce} from "@react-aria/live-announcer";
import {useDOMRef as $N9qJl$useDOMRef, useStyleProps as $N9qJl$useStyleProps, classNames as $N9qJl$classNames} from "@react-spectrum/utils";
import $N9qJl$spectrumiconsuiCrossLarge from "@spectrum-icons/ui/CrossLarge";
import {filterDOMProps as $N9qJl$filterDOMProps} from "@react-aria/utils";
import {FocusScope as $N9qJl$FocusScope} from "@react-aria/focus";
import {OpenTransition as $N9qJl$OpenTransition} from "@react-spectrum/overlays";
import $N9qJl$react, {useState as $N9qJl$useState, useRef as $N9qJl$useRef, useEffect as $N9qJl$useEffect} from "react";
import {Text as $N9qJl$Text} from "@react-spectrum/text";
import {useKeyboard as $N9qJl$useKeyboard} from "@react-aria/interactions";
import {useLocalizedStringFormatter as $N9qJl$useLocalizedStringFormatter} from "@react-aria/i18n";
import {useProviderProps as $N9qJl$useProviderProps} from "@react-spectrum/provider";

function $parcel$interopDefault(a) {
  return a && a.__esModule ? a.default : a;
}
function $parcel$export(e, n, v, s) {
  Object.defineProperty(e, n, {get: v, set: s, enumerable: true, configurable: true});
}
/*
 * Copyright 2020 Adobe. All rights reserved.
 * This file is licensed to you under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License. You may obtain a copy
 * of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under
 * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 * OF ANY KIND, either express or implied. See the License for the specific language
 * governing permissions and limitations under the License.
 */ /// <reference types="css-module-types" />
/*
 * Copyright 2020 Adobe. All rights reserved.
 * This file is licensed to you under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License. You may obtain a copy
 * of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under
 * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 * OF ANY KIND, either express or implied. See the License for the specific language
 * governing permissions and limitations under the License.
 */ 






var $29b198d669109c8b$exports = {};
var $42f1aaf0cd835ede$exports = {};
$42f1aaf0cd835ede$exports = {
    "actions": `الإجراءات`,
    "actionsAvailable": `الإجراءات المتاحة.`,
    "clearSelection": `إزالة التحديد`,
    "selected": (args, formatter)=>`${formatter.plural(args.count, {
            "=0": `غير محدد`,
            other: ()=>`${formatter.number(args.count)} محدد`
        })}`,
    "selectedAll": `تم تحديد الكل`
};


var $22c931b8c5fa9db9$exports = {};
$22c931b8c5fa9db9$exports = {
    "actions": `Действия`,
    "actionsAvailable": `Налични действия.`,
    "clearSelection": `Изчистване на избора`,
    "selected": (args, formatter)=>`${formatter.plural(args.count, {
            "=0": `Няма избрани`,
            other: ()=>`${formatter.number(args.count)} избрани`
        })}`,
    "selectedAll": `Всички избрани`
};


var $07dc24deaaf64ab0$exports = {};
$07dc24deaaf64ab0$exports = {
    "actions": `Akce`,
    "actionsAvailable": `Dostupné akce.`,
    "clearSelection": `Vymazat výběr`,
    "selected": (args, formatter)=>`${formatter.plural(args.count, {
            "=0": `Nic není vybráno`,
            other: ()=>`Vybráno ${formatter.number(args.count)}`
        })}`,
    "selectedAll": `Vybráno vše`
};


var $44fd3a75baaf9937$exports = {};
$44fd3a75baaf9937$exports = {
    "actions": `Handlinger`,
    "actionsAvailable": `Tilgængelige handlinger.`,
    "clearSelection": `Ryd markering`,
    "selected": (args, formatter)=>`${formatter.plural(args.count, {
            "=0": `Ingen valgt`,
            other: ()=>`${formatter.number(args.count)} valgt`
        })}`,
    "selectedAll": `Alle valgt`
};


var $3281990be7bd78f6$exports = {};
$3281990be7bd78f6$exports = {
    "actions": `Aktionen`,
    "actionsAvailable": `Aktionen verfügbar.`,
    "clearSelection": `Auswahl löschen`,
    "selected": (args, formatter)=>`${formatter.plural(args.count, {
            "=0": `Nichts ausgewählt`,
            other: ()=>`${formatter.number(args.count)} ausgewählt`
        })}`,
    "selectedAll": `Alles ausgewählt`
};


var $242cbd951f3bcfeb$exports = {};
$242cbd951f3bcfeb$exports = {
    "actions": `Ενέργειες`,
    "actionsAvailable": `Υπάρχουν διαθέσιμες ενέργειες.`,
    "clearSelection": `Εκκαθάριση επιλογής`,
    "selected": (args, formatter)=>`${formatter.plural(args.count, {
            "=0": `Δεν επιλέχθηκε κανένα`,
            other: ()=>`Επιλέχθηκαν ${formatter.number(args.count)}`
        })}`,
    "selectedAll": `Επιλέχθηκαν όλα`
};


var $117deeeb64547dc7$exports = {};
$117deeeb64547dc7$exports = {
    "clearSelection": `Clear selection`,
    "selected": (args, formatter)=>`${formatter.plural(args.count, {
            "=0": `None selected`,
            other: ()=>`${formatter.number(args.count)} selected`
        })}`,
    "selectedAll": `All selected`,
    "actions": `Actions`,
    "actionsAvailable": `Actions available.`
};


var $5b8605d86a9af2ee$exports = {};
$5b8605d86a9af2ee$exports = {
    "actions": `Acciones`,
    "actionsAvailable": `Acciones disponibles.`,
    "clearSelection": `Borrar selección`,
    "selected": (args, formatter)=>`${formatter.plural(args.count, {
            "=0": `Nada seleccionado`,
            other: ()=>`${formatter.number(args.count)} seleccionado`
        })}`,
    "selectedAll": `Todo seleccionado`
};


var $5530e5c9f2623025$exports = {};
$5530e5c9f2623025$exports = {
    "actions": `Toimingud`,
    "actionsAvailable": `Toimingud saadaval.`,
    "clearSelection": `Puhasta valik`,
    "selected": (args, formatter)=>`${formatter.plural(args.count, {
            "=0": `Pole valitud`,
            other: ()=>`${formatter.number(args.count)} valitud`
        })}`,
    "selectedAll": `Kõik valitud`
};


var $7a405d1dfc563131$exports = {};
$7a405d1dfc563131$exports = {
    "actions": `Toiminnot`,
    "actionsAvailable": `Toiminnot käytettävissä.`,
    "clearSelection": `Poista valinta`,
    "selected": (args, formatter)=>`${formatter.plural(args.count, {
            "=0": `Ei mitään valittu`,
            other: ()=>`${formatter.number(args.count)} valittu`
        })}`,
    "selectedAll": `Kaikki valittu`
};


var $e0c3da022513542b$exports = {};
$e0c3da022513542b$exports = {
    "actions": `Actions`,
    "actionsAvailable": `Actions disponibles.`,
    "clearSelection": `Supprimer la sélection`,
    "selected": (args, formatter)=>`${formatter.plural(args.count, {
            "=0": `Aucun élément sélectionné`,
            other: ()=>`${formatter.number(args.count)} sélectionnés`
        })}`,
    "selectedAll": `Toute la sélection`
};


var $796ff5aed019aee1$exports = {};
$796ff5aed019aee1$exports = {
    "actions": `פעולות`,
    "actionsAvailable": `פעולות זמינות.`,
    "clearSelection": `נקה בחירה`,
    "selected": (args, formatter)=>`${formatter.plural(args.count, {
            "=0": `לא בוצעה בחירה`,
            other: ()=>`${formatter.number(args.count)} נבחרו`
        })}`,
    "selectedAll": `כל הפריטים שנבחרו`
};


var $e1066348ea80fd36$exports = {};
$e1066348ea80fd36$exports = {
    "actions": `Radnje`,
    "actionsAvailable": `Dostupne radnje.`,
    "clearSelection": `Poništi odabir`,
    "selected": (args, formatter)=>`${formatter.plural(args.count, {
            "=0": `Nijedna nije odabrana`,
            other: ()=>`${formatter.number(args.count)} je odabrano`
        })}`,
    "selectedAll": `Sve je odabrano`
};


var $939ac78e6c54d005$exports = {};
$939ac78e6c54d005$exports = {
    "actions": `Műveletek`,
    "actionsAvailable": `Műveletek állnak rendelkezésre.`,
    "clearSelection": `Kijelölés törlése`,
    "selected": (args, formatter)=>`${formatter.plural(args.count, {
            "=0": `Egy sincs kijelölve`,
            other: ()=>`${formatter.number(args.count)} kijelölve`
        })}`,
    "selectedAll": `Mind kijelölve`
};


var $087af352bd8a67a5$exports = {};
$087af352bd8a67a5$exports = {
    "actions": `Azioni`,
    "actionsAvailable": `Azioni disponibili.`,
    "clearSelection": `Annulla selezione`,
    "selected": (args, formatter)=>`${formatter.plural(args.count, {
            "=0": `Nessuno selezionato`,
            other: ()=>`${formatter.number(args.count)} selezionato/i`
        })}`,
    "selectedAll": `Tutti selezionati`
};


var $ef25b75f84bf9640$exports = {};
$ef25b75f84bf9640$exports = {
    "actions": `アクション`,
    "actionsAvailable": `アクションを利用できます。`,
    "clearSelection": `選択をクリア`,
    "selected": (args, formatter)=>`${formatter.plural(args.count, {
            "=0": `選択されていません`,
            other: ()=>`${formatter.number(args.count)} 個を選択しました`
        })}`,
    "selectedAll": `すべてを選択`
};


var $a9fcaa28ab16ac58$exports = {};
$a9fcaa28ab16ac58$exports = {
    "actions": `액션`,
    "actionsAvailable": `사용 가능한 액션`,
    "clearSelection": `선택 항목 지우기`,
    "selected": (args, formatter)=>`${formatter.plural(args.count, {
            "=0": `선택된 항목 없음`,
            other: ()=>`${formatter.number(args.count)}개 선택됨`
        })}`,
    "selectedAll": `모두 선택됨`
};


var $7103396bd37d192c$exports = {};
$7103396bd37d192c$exports = {
    "actions": `Veiksmai`,
    "actionsAvailable": `Galimi veiksmai.`,
    "clearSelection": `Išvalyti pasirinkimą`,
    "selected": (args, formatter)=>`${formatter.plural(args.count, {
            "=0": `Nieko nepasirinkta`,
            other: ()=>`Pasirinkta ${formatter.number(args.count)}`
        })}`,
    "selectedAll": `Pasirinkta viskas`
};


var $da39587a937fb6fd$exports = {};
$da39587a937fb6fd$exports = {
    "actions": `Darbības`,
    "actionsAvailable": `Pieejamas darbības.`,
    "clearSelection": `Notīrīt atlasi`,
    "selected": (args, formatter)=>`${formatter.plural(args.count, {
            "=0": `Nav atlasīts nekas`,
            other: ()=>`Atlasīts: ${formatter.number(args.count)}`
        })}`,
    "selectedAll": `Atlasīts viss`
};


var $fef3c3414d151198$exports = {};
$fef3c3414d151198$exports = {
    "actions": `Handlinger`,
    "actionsAvailable": `Tilgjengelige handlinger.`,
    "clearSelection": `Tøm utvalg`,
    "selected": (args, formatter)=>`${formatter.plural(args.count, {
            "=0": `Ingen er valgt`,
            other: ()=>`${formatter.number(args.count)} er valgt`
        })}`,
    "selectedAll": `Alle er valgt`
};


var $3cfd427955fee95c$exports = {};
$3cfd427955fee95c$exports = {
    "actions": `Acties`,
    "actionsAvailable": `Acties beschikbaar.`,
    "clearSelection": `Selectie wissen`,
    "selected": (args, formatter)=>`${formatter.plural(args.count, {
            "=0": `Niets geselecteerd`,
            other: ()=>`${formatter.number(args.count)} geselecteerd`
        })}`,
    "selectedAll": `Alles geselecteerd`
};


var $ecd102793531599a$exports = {};
$ecd102793531599a$exports = {
    "actions": `Działania`,
    "actionsAvailable": `Dostępne działania.`,
    "clearSelection": `Wyczyść zaznaczenie`,
    "selected": (args, formatter)=>`${formatter.plural(args.count, {
            "=0": `Nie zaznaczono żadnego elementu`,
            other: ()=>`${formatter.number(args.count)} zaznaczonych`
        })}`,
    "selectedAll": `Wszystkie zaznaczone`
};


var $70b0a95af4186e29$exports = {};
$70b0a95af4186e29$exports = {
    "actions": `Ações`,
    "actionsAvailable": `Ações disponíveis.`,
    "clearSelection": `Limpar seleção`,
    "selected": (args, formatter)=>`${formatter.plural(args.count, {
            "=0": `Nada selecionado`,
            other: ()=>`${formatter.number(args.count)} selecionado`
        })}`,
    "selectedAll": `Todos selecionados`
};


var $4b15e180aed5c52d$exports = {};
$4b15e180aed5c52d$exports = {
    "actions": `Ações`,
    "actionsAvailable": `Ações disponíveis.`,
    "clearSelection": `Limpar seleção`,
    "selected": (args, formatter)=>`${formatter.plural(args.count, {
            "=0": `Nenhum selecionado`,
            other: ()=>`${formatter.number(args.count)} selecionado`
        })}`,
    "selectedAll": `Tudo selecionado`
};


var $d0ef0a1cef9c2d6c$exports = {};
$d0ef0a1cef9c2d6c$exports = {
    "actions": `Acțiuni`,
    "actionsAvailable": `Acțiuni disponibile.`,
    "clearSelection": `Goliți selecția`,
    "selected": (args, formatter)=>`${formatter.plural(args.count, {
            "=0": `Niciunul selectat`,
            other: ()=>`${formatter.number(args.count)} selectate`
        })}`,
    "selectedAll": `Toate selectate`
};


var $58fc1ff371f0c7fd$exports = {};
$58fc1ff371f0c7fd$exports = {
    "actions": `Действия`,
    "actionsAvailable": `Возможно выполнение действий.`,
    "clearSelection": `Очистить выбор`,
    "selected": (args, formatter)=>`${formatter.plural(args.count, {
            "=0": `Ничего не выбоано`,
            other: ()=>`${formatter.number(args.count)} выбрано`
        })}`,
    "selectedAll": `Выбрано все`
};


var $b5f65f4e7988d1dd$exports = {};
$b5f65f4e7988d1dd$exports = {
    "actions": `Akcie`,
    "actionsAvailable": `Dostupné akcie.`,
    "clearSelection": `Vymazať výber`,
    "selected": (args, formatter)=>`${formatter.plural(args.count, {
            "=0": `Žiadne vybraté položky`,
            other: ()=>`Počet vybratých položiek: ${formatter.number(args.count)}`
        })}`,
    "selectedAll": `Všetky vybraté položky`
};


var $5409ad409aeaf678$exports = {};
$5409ad409aeaf678$exports = {
    "actions": `Dejanja`,
    "actionsAvailable": `Na voljo so dejanja.`,
    "clearSelection": `Počisti izbor`,
    "selected": (args, formatter)=>`${formatter.plural(args.count, {
            "=0": `Nič izbranih`,
            other: ()=>`${formatter.number(args.count)} izbranih`
        })}`,
    "selectedAll": `Vsi izbrani`
};


var $43feb38b3d4bbf1f$exports = {};
$43feb38b3d4bbf1f$exports = {
    "actions": `Radnje`,
    "actionsAvailable": `Dostupne su radnje.`,
    "clearSelection": `Poništi izbor`,
    "selected": (args, formatter)=>`${formatter.plural(args.count, {
            "=0": `Ništa nije izabrano`,
            other: ()=>`${formatter.number(args.count)} je izabrano`
        })}`,
    "selectedAll": `Sve je izabrano`
};


var $cbe41ff823ebe108$exports = {};
$cbe41ff823ebe108$exports = {
    "actions": `Åtgärder`,
    "actionsAvailable": `Åtgärder finns.`,
    "clearSelection": `Rensa markering`,
    "selected": (args, formatter)=>`${formatter.plural(args.count, {
            "=0": `Inga markerade`,
            other: ()=>`${formatter.number(args.count)} markerade`
        })}`,
    "selectedAll": `Alla markerade`
};


var $57b680843fcbe09e$exports = {};
$57b680843fcbe09e$exports = {
    "actions": `Eylemler`,
    "actionsAvailable": `Eylemler mevcut.`,
    "clearSelection": `Seçimi temizle`,
    "selected": (args, formatter)=>`${formatter.plural(args.count, {
            "=0": `Hiçbiri seçilmedi`,
            other: ()=>`${formatter.number(args.count)} seçildi`
        })}`,
    "selectedAll": `Tümü seçildi`
};


var $1e4c96bc308e0da2$exports = {};
$1e4c96bc308e0da2$exports = {
    "actions": `Дії`,
    "actionsAvailable": `Доступні дії.`,
    "clearSelection": `Очистити вибір`,
    "selected": (args, formatter)=>`${formatter.plural(args.count, {
            "=0": `Нічого не вибрано`,
            other: ()=>`Вибрано: ${formatter.number(args.count)}`
        })}`,
    "selectedAll": `Усе вибрано`
};


var $18faa55a67dca8f3$exports = {};
$18faa55a67dca8f3$exports = {
    "actions": `操作`,
    "actionsAvailable": `有可用操作。`,
    "clearSelection": `清除选择`,
    "selected": (args, formatter)=>`${formatter.plural(args.count, {
            "=0": `无选择`,
            other: ()=>`已选择 ${formatter.number(args.count)} 个`
        })}`,
    "selectedAll": `全选`
};


var $804c64c525579787$exports = {};
$804c64c525579787$exports = {
    "actions": `動作`,
    "actionsAvailable": `可執行的動作。`,
    "clearSelection": `清除選取項目`,
    "selected": (args, formatter)=>`${formatter.plural(args.count, {
            "=0": `未選取任何項目`,
            other: ()=>`已選取 ${formatter.number(args.count)} 個`
        })}`,
    "selectedAll": `已選取所有項目`
};


$29b198d669109c8b$exports = {
    "ar-AE": $42f1aaf0cd835ede$exports,
    "bg-BG": $22c931b8c5fa9db9$exports,
    "cs-CZ": $07dc24deaaf64ab0$exports,
    "da-DK": $44fd3a75baaf9937$exports,
    "de-DE": $3281990be7bd78f6$exports,
    "el-GR": $242cbd951f3bcfeb$exports,
    "en-US": $117deeeb64547dc7$exports,
    "es-ES": $5b8605d86a9af2ee$exports,
    "et-EE": $5530e5c9f2623025$exports,
    "fi-FI": $7a405d1dfc563131$exports,
    "fr-FR": $e0c3da022513542b$exports,
    "he-IL": $796ff5aed019aee1$exports,
    "hr-HR": $e1066348ea80fd36$exports,
    "hu-HU": $939ac78e6c54d005$exports,
    "it-IT": $087af352bd8a67a5$exports,
    "ja-JP": $ef25b75f84bf9640$exports,
    "ko-KR": $a9fcaa28ab16ac58$exports,
    "lt-LT": $7103396bd37d192c$exports,
    "lv-LV": $da39587a937fb6fd$exports,
    "nb-NO": $fef3c3414d151198$exports,
    "nl-NL": $3cfd427955fee95c$exports,
    "pl-PL": $ecd102793531599a$exports,
    "pt-BR": $70b0a95af4186e29$exports,
    "pt-PT": $4b15e180aed5c52d$exports,
    "ro-RO": $d0ef0a1cef9c2d6c$exports,
    "ru-RU": $58fc1ff371f0c7fd$exports,
    "sk-SK": $b5f65f4e7988d1dd$exports,
    "sl-SI": $5409ad409aeaf678$exports,
    "sr-SP": $43feb38b3d4bbf1f$exports,
    "sv-SE": $cbe41ff823ebe108$exports,
    "tr-TR": $57b680843fcbe09e$exports,
    "uk-UA": $1e4c96bc308e0da2$exports,
    "zh-CN": $18faa55a67dca8f3$exports,
    "zh-TW": $804c64c525579787$exports
};




var $7bd9eeb2cd70bd0c$exports = {};

$parcel$export($7bd9eeb2cd70bd0c$exports, "ActionBarContainer", () => $7bd9eeb2cd70bd0c$export$ac2eb07f267e434c, (v) => $7bd9eeb2cd70bd0c$export$ac2eb07f267e434c = v);
$parcel$export($7bd9eeb2cd70bd0c$exports, "react-spectrum-ActionBar", () => $7bd9eeb2cd70bd0c$export$71a4c2be37bc4ee5, (v) => $7bd9eeb2cd70bd0c$export$71a4c2be37bc4ee5 = v);
$parcel$export($7bd9eeb2cd70bd0c$exports, "is-open", () => $7bd9eeb2cd70bd0c$export$a9781837241c946d, (v) => $7bd9eeb2cd70bd0c$export$a9781837241c946d = v);
$parcel$export($7bd9eeb2cd70bd0c$exports, "react-spectrum-ActionBar-bar", () => $7bd9eeb2cd70bd0c$export$ea5a1c79ca7998e9, (v) => $7bd9eeb2cd70bd0c$export$ea5a1c79ca7998e9 = v);
$parcel$export($7bd9eeb2cd70bd0c$exports, "react-spectrum-ActionBar--emphasized", () => $7bd9eeb2cd70bd0c$export$3377f5e3cddd68a7, (v) => $7bd9eeb2cd70bd0c$export$3377f5e3cddd68a7 = v);
$parcel$export($7bd9eeb2cd70bd0c$exports, "react-spectrum-ActionBar-selectedCount", () => $7bd9eeb2cd70bd0c$export$8b9fb52122670f9e, (v) => $7bd9eeb2cd70bd0c$export$8b9fb52122670f9e = v);
$parcel$export($7bd9eeb2cd70bd0c$exports, "react-spectrum-ActionBar-actionGroup", () => $7bd9eeb2cd70bd0c$export$adc2706661946993, (v) => $7bd9eeb2cd70bd0c$export$adc2706661946993 = v);
var $7bd9eeb2cd70bd0c$export$ac2eb07f267e434c;
var $7bd9eeb2cd70bd0c$export$71a4c2be37bc4ee5;
var $7bd9eeb2cd70bd0c$export$a9781837241c946d;
var $7bd9eeb2cd70bd0c$export$ea5a1c79ca7998e9;
var $7bd9eeb2cd70bd0c$export$3377f5e3cddd68a7;
var $7bd9eeb2cd70bd0c$export$8b9fb52122670f9e;
var $7bd9eeb2cd70bd0c$export$adc2706661946993;
$7bd9eeb2cd70bd0c$export$ac2eb07f267e434c = "ActionBarContainer_cca088";
$7bd9eeb2cd70bd0c$export$71a4c2be37bc4ee5 = "react-spectrum-ActionBar_cca088";
$7bd9eeb2cd70bd0c$export$a9781837241c946d = "is-open_cca088";
$7bd9eeb2cd70bd0c$export$ea5a1c79ca7998e9 = "react-spectrum-ActionBar-bar_cca088";
$7bd9eeb2cd70bd0c$export$3377f5e3cddd68a7 = "react-spectrum-ActionBar--emphasized_cca088";
$7bd9eeb2cd70bd0c$export$8b9fb52122670f9e = "react-spectrum-ActionBar-selectedCount_cca088";
$7bd9eeb2cd70bd0c$export$adc2706661946993 = "react-spectrum-ActionBar-actionGroup_cca088";






function $a4e665fbd087d71e$var$ActionBar(props, ref) {
    let isOpen = props.selectedItemCount !== 0;
    let domRef = (0, $N9qJl$useDOMRef)(ref);
    return /*#__PURE__*/ (0, $N9qJl$react).createElement((0, $N9qJl$OpenTransition), {
        nodeRef: domRef,
        in: isOpen,
        mountOnEnter: true,
        unmountOnExit: true
    }, /*#__PURE__*/ (0, $N9qJl$react).createElement($a4e665fbd087d71e$var$ActionBarInnerWithRef, {
        ...props,
        ref: domRef
    }));
}
function $a4e665fbd087d71e$var$ActionBarInner(props, ref) {
    props = (0, $N9qJl$useProviderProps)(props);
    let { children: children , isEmphasized: isEmphasized , onAction: onAction , onClearSelection: onClearSelection , selectedItemCount: selectedItemCount , isOpen: isOpen , items: items  } = props;
    let { styleProps: styleProps  } = (0, $N9qJl$useStyleProps)(props);
    let stringFormatter = (0, $N9qJl$useLocalizedStringFormatter)((0, (/*@__PURE__*/$parcel$interopDefault($29b198d669109c8b$exports))));
    // Store the last count greater than zero in a ref so that we can retain it while rendering the fade-out animation.
    let [lastCount, setLastCount] = (0, $N9qJl$useState)(selectedItemCount);
    if ((selectedItemCount === "all" || selectedItemCount > 0) && selectedItemCount !== lastCount) setLastCount(selectedItemCount);
    let { keyboardProps: keyboardProps  } = (0, $N9qJl$useKeyboard)({
        onKeyDown (e) {
            if (e.key === "Escape") {
                e.preventDefault();
                onClearSelection();
            }
        }
    });
    // Announce "actions available" on mount.
    let isInitial = (0, $N9qJl$useRef)(true);
    (0, $N9qJl$useEffect)(()=>{
        if (isInitial.current) {
            isInitial.current = false;
            (0, $N9qJl$announce)(stringFormatter.format("actionsAvailable"));
        }
    }, [
        stringFormatter
    ]);
    return /*#__PURE__*/ (0, $N9qJl$react).createElement((0, $N9qJl$FocusScope), {
        restoreFocus: true
    }, /*#__PURE__*/ (0, $N9qJl$react).createElement("div", {
        ...(0, $N9qJl$filterDOMProps)(props),
        ...styleProps,
        ...keyboardProps,
        ref: ref,
        className: (0, $N9qJl$classNames)((0, (/*@__PURE__*/$parcel$interopDefault($7bd9eeb2cd70bd0c$exports))), "react-spectrum-ActionBar", {
            "react-spectrum-ActionBar--emphasized": isEmphasized,
            "is-open": isOpen
        }, styleProps.className)
    }, /*#__PURE__*/ (0, $N9qJl$react).createElement("div", {
        className: (0, $N9qJl$classNames)((0, (/*@__PURE__*/$parcel$interopDefault($7bd9eeb2cd70bd0c$exports))), "react-spectrum-ActionBar-bar")
    }, /*#__PURE__*/ (0, $N9qJl$react).createElement((0, $N9qJl$ActionGroup), {
        items: items,
        "aria-label": stringFormatter.format("actions"),
        isQuiet: true,
        staticColor: isEmphasized ? "white" : undefined,
        overflowMode: "collapse",
        buttonLabelBehavior: "collapse",
        onAction: onAction,
        UNSAFE_className: (0, $N9qJl$classNames)((0, (/*@__PURE__*/$parcel$interopDefault($7bd9eeb2cd70bd0c$exports))), "react-spectrum-ActionBar-actionGroup")
    }, children), /*#__PURE__*/ (0, $N9qJl$react).createElement((0, $N9qJl$ActionButton), {
        gridArea: "clear",
        "aria-label": stringFormatter.format("clearSelection"),
        onPress: ()=>onClearSelection(),
        isQuiet: true,
        staticColor: isEmphasized ? "white" : undefined
    }, /*#__PURE__*/ (0, $N9qJl$react).createElement((0, $N9qJl$spectrumiconsuiCrossLarge), null)), /*#__PURE__*/ (0, $N9qJl$react).createElement((0, $N9qJl$Text), {
        UNSAFE_className: (0, $N9qJl$classNames)((0, (/*@__PURE__*/$parcel$interopDefault($7bd9eeb2cd70bd0c$exports))), "react-spectrum-ActionBar-selectedCount")
    }, lastCount === "all" ? stringFormatter.format("selectedAll") : stringFormatter.format("selected", {
        count: lastCount
    })))));
}
const $a4e665fbd087d71e$var$ActionBarInnerWithRef = /*#__PURE__*/ (0, $N9qJl$react).forwardRef($a4e665fbd087d71e$var$ActionBarInner);
/**
 * Action bars are used for single and bulk selection patterns when a user needs to perform actions on one or more items at the same time.
 */ const $a4e665fbd087d71e$export$e213cebad6250b4a = /*#__PURE__*/ (0, $N9qJl$react).forwardRef($a4e665fbd087d71e$var$ActionBar);


/*
 * Copyright 2020 Adobe. All rights reserved.
 * This file is licensed to you under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License. You may obtain a copy
 * of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under
 * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 * OF ANY KIND, either express or implied. See the License for the specific language
 * governing permissions and limitations under the License.
 */ 




function $2f6c8b20aa071c6d$var$ActionBarContainer(props, ref) {
    // Grabs specific props from the closest Provider (see https://react-spectrum.adobe.com/react-spectrum/Provider.html#property-groups). Remove if your component doesn't support any of the listed props.
    props = (0, $N9qJl$useProviderProps)(props);
    let { children: children  } = props;
    let { styleProps: styleProps  } = (0, $N9qJl$useStyleProps)(props);
    let domRef = (0, $N9qJl$useDOMRef)(ref);
    return /*#__PURE__*/ (0, $N9qJl$react).createElement("div", {
        ...(0, $N9qJl$filterDOMProps)(props),
        ...styleProps,
        ref: domRef,
        className: (0, $N9qJl$classNames)((0, (/*@__PURE__*/$parcel$interopDefault($7bd9eeb2cd70bd0c$exports))), "ActionBarContainer", styleProps.className)
    }, children);
}
/**
 * ActionBarContainer wraps around an ActionBar and a component that supports selection. It handles
 * the ActionBar's position with respect to its linked component.
 */ const $2f6c8b20aa071c6d$export$ac2eb07f267e434c = /*#__PURE__*/ (0, $N9qJl$react).forwardRef($2f6c8b20aa071c6d$var$ActionBarContainer);





export {$a4e665fbd087d71e$export$e213cebad6250b4a as ActionBar, $2f6c8b20aa071c6d$export$ac2eb07f267e434c as ActionBarContainer, $da70baa779e17e7a$re_export$Item as Item};
