// @react-spectrum/actionbar/dist/main.js
require("./main.css");
var $grxqE$reactstatelycollections = require("@react-stately/collections");
var $grxqE$reactspectrumbutton = require("@react-spectrum/button");
var $grxqE$reactspectrumactiongroup = require("@react-spectrum/actiongroup");
var $grxqE$reactarialiveannouncer = require("@react-aria/live-announcer");
var $grxqE$reactspectrumutils = require("@react-spectrum/utils");
var $grxqE$spectrumiconsuiCrossLarge = require("@spectrum-icons/ui/CrossLarge");
var $grxqE$reactariautils = require("@react-aria/utils");
var $grxqE$reactariafocus = require("@react-aria/focus");
var $grxqE$reactspectrumoverlays = require("@react-spectrum/overlays");
var $grxqE$react = require("react");
var $grxqE$reactspectrumtext = require("@react-spectrum/text");
var $grxqE$reactariainteractions = require("@react-aria/interactions");
var $grxqE$reactariai18n = require("@react-aria/i18n");
var $grxqE$reactspectrumprovider = require("@react-spectrum/provider");

function $parcel$export(e, n, v, s) {
  Object.defineProperty(e, n, {get: v, set: s, enumerable: true, configurable: true});
}
function $parcel$interopDefault(a) {
  return a && a.__esModule ? a.default : a;
}

$parcel$export(module.exports, "ActionBar", () => $7b5467653feb1153$export$e213cebad6250b4a);
$parcel$export(module.exports, "ActionBarContainer", () => $961b22d2b8bd2da1$export$ac2eb07f267e434c);
$parcel$export(module.exports, "Item", () => $grxqE$reactstatelycollections.Item);
/*
 * Copyright 2020 Adobe. All rights reserved.
 * This file is licensed to you under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License. You may obtain a copy
 * of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under
 * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 * OF ANY KIND, either express or implied. See the License for the specific language
 * governing permissions and limitations under the License.
 */ /// <reference types="css-module-types" />
/*
 * Copyright 2020 Adobe. All rights reserved.
 * This file is licensed to you under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License. You may obtain a copy
 * of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under
 * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 * OF ANY KIND, either express or implied. See the License for the specific language
 * governing permissions and limitations under the License.
 */ 






var $b1ba52ca91e22687$exports = {};
var $a42f3d97a30aa6c9$exports = {};
$a42f3d97a30aa6c9$exports = {
    "actions": `الإجراءات`,
    "actionsAvailable": `الإجراءات المتاحة.`,
    "clearSelection": `إزالة التحديد`,
    "selected": (args, formatter)=>`${formatter.plural(args.count, {
            "=0": `غير محدد`,
            other: ()=>`${formatter.number(args.count)} محدد`
        })}`,
    "selectedAll": `تم تحديد الكل`
};


var $527a50f3b3c911b7$exports = {};
$527a50f3b3c911b7$exports = {
    "actions": `Действия`,
    "actionsAvailable": `Налични действия.`,
    "clearSelection": `Изчистване на избора`,
    "selected": (args, formatter)=>`${formatter.plural(args.count, {
            "=0": `Няма избрани`,
            other: ()=>`${formatter.number(args.count)} избрани`
        })}`,
    "selectedAll": `Всички избрани`
};


var $cd90476fdc303242$exports = {};
$cd90476fdc303242$exports = {
    "actions": `Akce`,
    "actionsAvailable": `Dostupné akce.`,
    "clearSelection": `Vymazat výběr`,
    "selected": (args, formatter)=>`${formatter.plural(args.count, {
            "=0": `Nic není vybráno`,
            other: ()=>`Vybráno ${formatter.number(args.count)}`
        })}`,
    "selectedAll": `Vybráno vše`
};


var $081b7298af5f5b69$exports = {};
$081b7298af5f5b69$exports = {
    "actions": `Handlinger`,
    "actionsAvailable": `Tilgængelige handlinger.`,
    "clearSelection": `Ryd markering`,
    "selected": (args, formatter)=>`${formatter.plural(args.count, {
            "=0": `Ingen valgt`,
            other: ()=>`${formatter.number(args.count)} valgt`
        })}`,
    "selectedAll": `Alle valgt`
};


var $6ef7dffbce87e74f$exports = {};
$6ef7dffbce87e74f$exports = {
    "actions": `Aktionen`,
    "actionsAvailable": `Aktionen verfügbar.`,
    "clearSelection": `Auswahl löschen`,
    "selected": (args, formatter)=>`${formatter.plural(args.count, {
            "=0": `Nichts ausgewählt`,
            other: ()=>`${formatter.number(args.count)} ausgewählt`
        })}`,
    "selectedAll": `Alles ausgewählt`
};


var $43c62672ca45763f$exports = {};
$43c62672ca45763f$exports = {
    "actions": `Ενέργειες`,
    "actionsAvailable": `Υπάρχουν διαθέσιμες ενέργειες.`,
    "clearSelection": `Εκκαθάριση επιλογής`,
    "selected": (args, formatter)=>`${formatter.plural(args.count, {
            "=0": `Δεν επιλέχθηκε κανένα`,
            other: ()=>`Επιλέχθηκαν ${formatter.number(args.count)}`
        })}`,
    "selectedAll": `Επιλέχθηκαν όλα`
};


var $4a93342b52e9ddcf$exports = {};
$4a93342b52e9ddcf$exports = {
    "clearSelection": `Clear selection`,
    "selected": (args, formatter)=>`${formatter.plural(args.count, {
            "=0": `None selected`,
            other: ()=>`${formatter.number(args.count)} selected`
        })}`,
    "selectedAll": `All selected`,
    "actions": `Actions`,
    "actionsAvailable": `Actions available.`
};


var $d1fe0a28d16ba803$exports = {};
$d1fe0a28d16ba803$exports = {
    "actions": `Acciones`,
    "actionsAvailable": `Acciones disponibles.`,
    "clearSelection": `Borrar selección`,
    "selected": (args, formatter)=>`${formatter.plural(args.count, {
            "=0": `Nada seleccionado`,
            other: ()=>`${formatter.number(args.count)} seleccionado`
        })}`,
    "selectedAll": `Todo seleccionado`
};


var $3f607057b4497412$exports = {};
$3f607057b4497412$exports = {
    "actions": `Toimingud`,
    "actionsAvailable": `Toimingud saadaval.`,
    "clearSelection": `Puhasta valik`,
    "selected": (args, formatter)=>`${formatter.plural(args.count, {
            "=0": `Pole valitud`,
            other: ()=>`${formatter.number(args.count)} valitud`
        })}`,
    "selectedAll": `Kõik valitud`
};


var $dffc096b505cd1a5$exports = {};
$dffc096b505cd1a5$exports = {
    "actions": `Toiminnot`,
    "actionsAvailable": `Toiminnot käytettävissä.`,
    "clearSelection": `Poista valinta`,
    "selected": (args, formatter)=>`${formatter.plural(args.count, {
            "=0": `Ei mitään valittu`,
            other: ()=>`${formatter.number(args.count)} valittu`
        })}`,
    "selectedAll": `Kaikki valittu`
};


var $fced00dcdde4e781$exports = {};
$fced00dcdde4e781$exports = {
    "actions": `Actions`,
    "actionsAvailable": `Actions disponibles.`,
    "clearSelection": `Supprimer la sélection`,
    "selected": (args, formatter)=>`${formatter.plural(args.count, {
            "=0": `Aucun élément sélectionné`,
            other: ()=>`${formatter.number(args.count)} sélectionnés`
        })}`,
    "selectedAll": `Toute la sélection`
};


var $e568cd7cb63d7ce3$exports = {};
$e568cd7cb63d7ce3$exports = {
    "actions": `פעולות`,
    "actionsAvailable": `פעולות זמינות.`,
    "clearSelection": `נקה בחירה`,
    "selected": (args, formatter)=>`${formatter.plural(args.count, {
            "=0": `לא בוצעה בחירה`,
            other: ()=>`${formatter.number(args.count)} נבחרו`
        })}`,
    "selectedAll": `כל הפריטים שנבחרו`
};


var $50951315acf49d05$exports = {};
$50951315acf49d05$exports = {
    "actions": `Radnje`,
    "actionsAvailable": `Dostupne radnje.`,
    "clearSelection": `Poništi odabir`,
    "selected": (args, formatter)=>`${formatter.plural(args.count, {
            "=0": `Nijedna nije odabrana`,
            other: ()=>`${formatter.number(args.count)} je odabrano`
        })}`,
    "selectedAll": `Sve je odabrano`
};


var $fd06dc9cc87d0258$exports = {};
$fd06dc9cc87d0258$exports = {
    "actions": `Műveletek`,
    "actionsAvailable": `Műveletek állnak rendelkezésre.`,
    "clearSelection": `Kijelölés törlése`,
    "selected": (args, formatter)=>`${formatter.plural(args.count, {
            "=0": `Egy sincs kijelölve`,
            other: ()=>`${formatter.number(args.count)} kijelölve`
        })}`,
    "selectedAll": `Mind kijelölve`
};


var $9bd9ce3c11c5d11e$exports = {};
$9bd9ce3c11c5d11e$exports = {
    "actions": `Azioni`,
    "actionsAvailable": `Azioni disponibili.`,
    "clearSelection": `Annulla selezione`,
    "selected": (args, formatter)=>`${formatter.plural(args.count, {
            "=0": `Nessuno selezionato`,
            other: ()=>`${formatter.number(args.count)} selezionato/i`
        })}`,
    "selectedAll": `Tutti selezionati`
};


var $9f76d796bf707b10$exports = {};
$9f76d796bf707b10$exports = {
    "actions": `アクション`,
    "actionsAvailable": `アクションを利用できます。`,
    "clearSelection": `選択をクリア`,
    "selected": (args, formatter)=>`${formatter.plural(args.count, {
            "=0": `選択されていません`,
            other: ()=>`${formatter.number(args.count)} 個を選択しました`
        })}`,
    "selectedAll": `すべてを選択`
};


var $94efe7bbbb291ef4$exports = {};
$94efe7bbbb291ef4$exports = {
    "actions": `액션`,
    "actionsAvailable": `사용 가능한 액션`,
    "clearSelection": `선택 항목 지우기`,
    "selected": (args, formatter)=>`${formatter.plural(args.count, {
            "=0": `선택된 항목 없음`,
            other: ()=>`${formatter.number(args.count)}개 선택됨`
        })}`,
    "selectedAll": `모두 선택됨`
};


var $92a06c1a52267a65$exports = {};
$92a06c1a52267a65$exports = {
    "actions": `Veiksmai`,
    "actionsAvailable": `Galimi veiksmai.`,
    "clearSelection": `Išvalyti pasirinkimą`,
    "selected": (args, formatter)=>`${formatter.plural(args.count, {
            "=0": `Nieko nepasirinkta`,
            other: ()=>`Pasirinkta ${formatter.number(args.count)}`
        })}`,
    "selectedAll": `Pasirinkta viskas`
};


var $670c23292e727d95$exports = {};
$670c23292e727d95$exports = {
    "actions": `Darbības`,
    "actionsAvailable": `Pieejamas darbības.`,
    "clearSelection": `Notīrīt atlasi`,
    "selected": (args, formatter)=>`${formatter.plural(args.count, {
            "=0": `Nav atlasīts nekas`,
            other: ()=>`Atlasīts: ${formatter.number(args.count)}`
        })}`,
    "selectedAll": `Atlasīts viss`
};


var $2636117ad082d1ea$exports = {};
$2636117ad082d1ea$exports = {
    "actions": `Handlinger`,
    "actionsAvailable": `Tilgjengelige handlinger.`,
    "clearSelection": `Tøm utvalg`,
    "selected": (args, formatter)=>`${formatter.plural(args.count, {
            "=0": `Ingen er valgt`,
            other: ()=>`${formatter.number(args.count)} er valgt`
        })}`,
    "selectedAll": `Alle er valgt`
};


var $e4ccf12b3bee45db$exports = {};
$e4ccf12b3bee45db$exports = {
    "actions": `Acties`,
    "actionsAvailable": `Acties beschikbaar.`,
    "clearSelection": `Selectie wissen`,
    "selected": (args, formatter)=>`${formatter.plural(args.count, {
            "=0": `Niets geselecteerd`,
            other: ()=>`${formatter.number(args.count)} geselecteerd`
        })}`,
    "selectedAll": `Alles geselecteerd`
};


var $b259f128f72a6ac6$exports = {};
$b259f128f72a6ac6$exports = {
    "actions": `Działania`,
    "actionsAvailable": `Dostępne działania.`,
    "clearSelection": `Wyczyść zaznaczenie`,
    "selected": (args, formatter)=>`${formatter.plural(args.count, {
            "=0": `Nie zaznaczono żadnego elementu`,
            other: ()=>`${formatter.number(args.count)} zaznaczonych`
        })}`,
    "selectedAll": `Wszystkie zaznaczone`
};


var $edd692d35256da76$exports = {};
$edd692d35256da76$exports = {
    "actions": `Ações`,
    "actionsAvailable": `Ações disponíveis.`,
    "clearSelection": `Limpar seleção`,
    "selected": (args, formatter)=>`${formatter.plural(args.count, {
            "=0": `Nada selecionado`,
            other: ()=>`${formatter.number(args.count)} selecionado`
        })}`,
    "selectedAll": `Todos selecionados`
};


var $0af50a83741a7f54$exports = {};
$0af50a83741a7f54$exports = {
    "actions": `Ações`,
    "actionsAvailable": `Ações disponíveis.`,
    "clearSelection": `Limpar seleção`,
    "selected": (args, formatter)=>`${formatter.plural(args.count, {
            "=0": `Nenhum selecionado`,
            other: ()=>`${formatter.number(args.count)} selecionado`
        })}`,
    "selectedAll": `Tudo selecionado`
};


var $1b60e4cd5d1e5b6b$exports = {};
$1b60e4cd5d1e5b6b$exports = {
    "actions": `Acțiuni`,
    "actionsAvailable": `Acțiuni disponibile.`,
    "clearSelection": `Goliți selecția`,
    "selected": (args, formatter)=>`${formatter.plural(args.count, {
            "=0": `Niciunul selectat`,
            other: ()=>`${formatter.number(args.count)} selectate`
        })}`,
    "selectedAll": `Toate selectate`
};


var $e9ca6eba44abffce$exports = {};
$e9ca6eba44abffce$exports = {
    "actions": `Действия`,
    "actionsAvailable": `Возможно выполнение действий.`,
    "clearSelection": `Очистить выбор`,
    "selected": (args, formatter)=>`${formatter.plural(args.count, {
            "=0": `Ничего не выбоано`,
            other: ()=>`${formatter.number(args.count)} выбрано`
        })}`,
    "selectedAll": `Выбрано все`
};


var $508bfd1ac48eb61a$exports = {};
$508bfd1ac48eb61a$exports = {
    "actions": `Akcie`,
    "actionsAvailable": `Dostupné akcie.`,
    "clearSelection": `Vymazať výber`,
    "selected": (args, formatter)=>`${formatter.plural(args.count, {
            "=0": `Žiadne vybraté položky`,
            other: ()=>`Počet vybratých položiek: ${formatter.number(args.count)}`
        })}`,
    "selectedAll": `Všetky vybraté položky`
};


var $d785f6c5ba148ce4$exports = {};
$d785f6c5ba148ce4$exports = {
    "actions": `Dejanja`,
    "actionsAvailable": `Na voljo so dejanja.`,
    "clearSelection": `Počisti izbor`,
    "selected": (args, formatter)=>`${formatter.plural(args.count, {
            "=0": `Nič izbranih`,
            other: ()=>`${formatter.number(args.count)} izbranih`
        })}`,
    "selectedAll": `Vsi izbrani`
};


var $dfc7dc8ff40cea70$exports = {};
$dfc7dc8ff40cea70$exports = {
    "actions": `Radnje`,
    "actionsAvailable": `Dostupne su radnje.`,
    "clearSelection": `Poništi izbor`,
    "selected": (args, formatter)=>`${formatter.plural(args.count, {
            "=0": `Ništa nije izabrano`,
            other: ()=>`${formatter.number(args.count)} je izabrano`
        })}`,
    "selectedAll": `Sve je izabrano`
};


var $bea42116379eb09f$exports = {};
$bea42116379eb09f$exports = {
    "actions": `Åtgärder`,
    "actionsAvailable": `Åtgärder finns.`,
    "clearSelection": `Rensa markering`,
    "selected": (args, formatter)=>`${formatter.plural(args.count, {
            "=0": `Inga markerade`,
            other: ()=>`${formatter.number(args.count)} markerade`
        })}`,
    "selectedAll": `Alla markerade`
};


var $dc03da5d819982a5$exports = {};
$dc03da5d819982a5$exports = {
    "actions": `Eylemler`,
    "actionsAvailable": `Eylemler mevcut.`,
    "clearSelection": `Seçimi temizle`,
    "selected": (args, formatter)=>`${formatter.plural(args.count, {
            "=0": `Hiçbiri seçilmedi`,
            other: ()=>`${formatter.number(args.count)} seçildi`
        })}`,
    "selectedAll": `Tümü seçildi`
};


var $c93c67a8485a02db$exports = {};
$c93c67a8485a02db$exports = {
    "actions": `Дії`,
    "actionsAvailable": `Доступні дії.`,
    "clearSelection": `Очистити вибір`,
    "selected": (args, formatter)=>`${formatter.plural(args.count, {
            "=0": `Нічого не вибрано`,
            other: ()=>`Вибрано: ${formatter.number(args.count)}`
        })}`,
    "selectedAll": `Усе вибрано`
};


var $dd336a8fdfa9aa7b$exports = {};
$dd336a8fdfa9aa7b$exports = {
    "actions": `操作`,
    "actionsAvailable": `有可用操作。`,
    "clearSelection": `清除选择`,
    "selected": (args, formatter)=>`${formatter.plural(args.count, {
            "=0": `无选择`,
            other: ()=>`已选择 ${formatter.number(args.count)} 个`
        })}`,
    "selectedAll": `全选`
};


var $30d1241741295d61$exports = {};
$30d1241741295d61$exports = {
    "actions": `動作`,
    "actionsAvailable": `可執行的動作。`,
    "clearSelection": `清除選取項目`,
    "selected": (args, formatter)=>`${formatter.plural(args.count, {
            "=0": `未選取任何項目`,
            other: ()=>`已選取 ${formatter.number(args.count)} 個`
        })}`,
    "selectedAll": `已選取所有項目`
};


$b1ba52ca91e22687$exports = {
    "ar-AE": $a42f3d97a30aa6c9$exports,
    "bg-BG": $527a50f3b3c911b7$exports,
    "cs-CZ": $cd90476fdc303242$exports,
    "da-DK": $081b7298af5f5b69$exports,
    "de-DE": $6ef7dffbce87e74f$exports,
    "el-GR": $43c62672ca45763f$exports,
    "en-US": $4a93342b52e9ddcf$exports,
    "es-ES": $d1fe0a28d16ba803$exports,
    "et-EE": $3f607057b4497412$exports,
    "fi-FI": $dffc096b505cd1a5$exports,
    "fr-FR": $fced00dcdde4e781$exports,
    "he-IL": $e568cd7cb63d7ce3$exports,
    "hr-HR": $50951315acf49d05$exports,
    "hu-HU": $fd06dc9cc87d0258$exports,
    "it-IT": $9bd9ce3c11c5d11e$exports,
    "ja-JP": $9f76d796bf707b10$exports,
    "ko-KR": $94efe7bbbb291ef4$exports,
    "lt-LT": $92a06c1a52267a65$exports,
    "lv-LV": $670c23292e727d95$exports,
    "nb-NO": $2636117ad082d1ea$exports,
    "nl-NL": $e4ccf12b3bee45db$exports,
    "pl-PL": $b259f128f72a6ac6$exports,
    "pt-BR": $edd692d35256da76$exports,
    "pt-PT": $0af50a83741a7f54$exports,
    "ro-RO": $1b60e4cd5d1e5b6b$exports,
    "ru-RU": $e9ca6eba44abffce$exports,
    "sk-SK": $508bfd1ac48eb61a$exports,
    "sl-SI": $d785f6c5ba148ce4$exports,
    "sr-SP": $dfc7dc8ff40cea70$exports,
    "sv-SE": $bea42116379eb09f$exports,
    "tr-TR": $dc03da5d819982a5$exports,
    "uk-UA": $c93c67a8485a02db$exports,
    "zh-CN": $dd336a8fdfa9aa7b$exports,
    "zh-TW": $30d1241741295d61$exports
};




var $ff24c7bbe0c168db$exports = {};

$parcel$export($ff24c7bbe0c168db$exports, "ActionBarContainer", () => $ff24c7bbe0c168db$export$ac2eb07f267e434c, (v) => $ff24c7bbe0c168db$export$ac2eb07f267e434c = v);
$parcel$export($ff24c7bbe0c168db$exports, "react-spectrum-ActionBar", () => $ff24c7bbe0c168db$export$71a4c2be37bc4ee5, (v) => $ff24c7bbe0c168db$export$71a4c2be37bc4ee5 = v);
$parcel$export($ff24c7bbe0c168db$exports, "is-open", () => $ff24c7bbe0c168db$export$a9781837241c946d, (v) => $ff24c7bbe0c168db$export$a9781837241c946d = v);
$parcel$export($ff24c7bbe0c168db$exports, "react-spectrum-ActionBar-bar", () => $ff24c7bbe0c168db$export$ea5a1c79ca7998e9, (v) => $ff24c7bbe0c168db$export$ea5a1c79ca7998e9 = v);
$parcel$export($ff24c7bbe0c168db$exports, "react-spectrum-ActionBar--emphasized", () => $ff24c7bbe0c168db$export$3377f5e3cddd68a7, (v) => $ff24c7bbe0c168db$export$3377f5e3cddd68a7 = v);
$parcel$export($ff24c7bbe0c168db$exports, "react-spectrum-ActionBar-selectedCount", () => $ff24c7bbe0c168db$export$8b9fb52122670f9e, (v) => $ff24c7bbe0c168db$export$8b9fb52122670f9e = v);
$parcel$export($ff24c7bbe0c168db$exports, "react-spectrum-ActionBar-actionGroup", () => $ff24c7bbe0c168db$export$adc2706661946993, (v) => $ff24c7bbe0c168db$export$adc2706661946993 = v);
var $ff24c7bbe0c168db$export$ac2eb07f267e434c;
var $ff24c7bbe0c168db$export$71a4c2be37bc4ee5;
var $ff24c7bbe0c168db$export$a9781837241c946d;
var $ff24c7bbe0c168db$export$ea5a1c79ca7998e9;
var $ff24c7bbe0c168db$export$3377f5e3cddd68a7;
var $ff24c7bbe0c168db$export$8b9fb52122670f9e;
var $ff24c7bbe0c168db$export$adc2706661946993;
$ff24c7bbe0c168db$export$ac2eb07f267e434c = "ActionBarContainer_cca088";
$ff24c7bbe0c168db$export$71a4c2be37bc4ee5 = "react-spectrum-ActionBar_cca088";
$ff24c7bbe0c168db$export$a9781837241c946d = "is-open_cca088";
$ff24c7bbe0c168db$export$ea5a1c79ca7998e9 = "react-spectrum-ActionBar-bar_cca088";
$ff24c7bbe0c168db$export$3377f5e3cddd68a7 = "react-spectrum-ActionBar--emphasized_cca088";
$ff24c7bbe0c168db$export$8b9fb52122670f9e = "react-spectrum-ActionBar-selectedCount_cca088";
$ff24c7bbe0c168db$export$adc2706661946993 = "react-spectrum-ActionBar-actionGroup_cca088";






function $7b5467653feb1153$var$ActionBar(props, ref) {
    let isOpen = props.selectedItemCount !== 0;
    let domRef = (0, $grxqE$reactspectrumutils.useDOMRef)(ref);
    return /*#__PURE__*/ (0, ($parcel$interopDefault($grxqE$react))).createElement((0, $grxqE$reactspectrumoverlays.OpenTransition), {
        nodeRef: domRef,
        in: isOpen,
        mountOnEnter: true,
        unmountOnExit: true
    }, /*#__PURE__*/ (0, ($parcel$interopDefault($grxqE$react))).createElement($7b5467653feb1153$var$ActionBarInnerWithRef, {
        ...props,
        ref: domRef
    }));
}
function $7b5467653feb1153$var$ActionBarInner(props, ref) {
    props = (0, $grxqE$reactspectrumprovider.useProviderProps)(props);
    let { children: children , isEmphasized: isEmphasized , onAction: onAction , onClearSelection: onClearSelection , selectedItemCount: selectedItemCount , isOpen: isOpen , items: items  } = props;
    let { styleProps: styleProps  } = (0, $grxqE$reactspectrumutils.useStyleProps)(props);
    let stringFormatter = (0, $grxqE$reactariai18n.useLocalizedStringFormatter)((0, (/*@__PURE__*/$parcel$interopDefault($b1ba52ca91e22687$exports))));
    // Store the last count greater than zero in a ref so that we can retain it while rendering the fade-out animation.
    let [lastCount, setLastCount] = (0, $grxqE$react.useState)(selectedItemCount);
    if ((selectedItemCount === "all" || selectedItemCount > 0) && selectedItemCount !== lastCount) setLastCount(selectedItemCount);
    let { keyboardProps: keyboardProps  } = (0, $grxqE$reactariainteractions.useKeyboard)({
        onKeyDown (e) {
            if (e.key === "Escape") {
                e.preventDefault();
                onClearSelection();
            }
        }
    });
    // Announce "actions available" on mount.
    let isInitial = (0, $grxqE$react.useRef)(true);
    (0, $grxqE$react.useEffect)(()=>{
        if (isInitial.current) {
            isInitial.current = false;
            (0, $grxqE$reactarialiveannouncer.announce)(stringFormatter.format("actionsAvailable"));
        }
    }, [
        stringFormatter
    ]);
    return /*#__PURE__*/ (0, ($parcel$interopDefault($grxqE$react))).createElement((0, $grxqE$reactariafocus.FocusScope), {
        restoreFocus: true
    }, /*#__PURE__*/ (0, ($parcel$interopDefault($grxqE$react))).createElement("div", {
        ...(0, $grxqE$reactariautils.filterDOMProps)(props),
        ...styleProps,
        ...keyboardProps,
        ref: ref,
        className: (0, $grxqE$reactspectrumutils.classNames)((0, (/*@__PURE__*/$parcel$interopDefault($ff24c7bbe0c168db$exports))), "react-spectrum-ActionBar", {
            "react-spectrum-ActionBar--emphasized": isEmphasized,
            "is-open": isOpen
        }, styleProps.className)
    }, /*#__PURE__*/ (0, ($parcel$interopDefault($grxqE$react))).createElement("div", {
        className: (0, $grxqE$reactspectrumutils.classNames)((0, (/*@__PURE__*/$parcel$interopDefault($ff24c7bbe0c168db$exports))), "react-spectrum-ActionBar-bar")
    }, /*#__PURE__*/ (0, ($parcel$interopDefault($grxqE$react))).createElement((0, $grxqE$reactspectrumactiongroup.ActionGroup), {
        items: items,
        "aria-label": stringFormatter.format("actions"),
        isQuiet: true,
        staticColor: isEmphasized ? "white" : undefined,
        overflowMode: "collapse",
        buttonLabelBehavior: "collapse",
        onAction: onAction,
        UNSAFE_className: (0, $grxqE$reactspectrumutils.classNames)((0, (/*@__PURE__*/$parcel$interopDefault($ff24c7bbe0c168db$exports))), "react-spectrum-ActionBar-actionGroup")
    }, children), /*#__PURE__*/ (0, ($parcel$interopDefault($grxqE$react))).createElement((0, $grxqE$reactspectrumbutton.ActionButton), {
        gridArea: "clear",
        "aria-label": stringFormatter.format("clearSelection"),
        onPress: ()=>onClearSelection(),
        isQuiet: true,
        staticColor: isEmphasized ? "white" : undefined
    }, /*#__PURE__*/ (0, ($parcel$interopDefault($grxqE$react))).createElement((0, ($parcel$interopDefault($grxqE$spectrumiconsuiCrossLarge))), null)), /*#__PURE__*/ (0, ($parcel$interopDefault($grxqE$react))).createElement((0, $grxqE$reactspectrumtext.Text), {
        UNSAFE_className: (0, $grxqE$reactspectrumutils.classNames)((0, (/*@__PURE__*/$parcel$interopDefault($ff24c7bbe0c168db$exports))), "react-spectrum-ActionBar-selectedCount")
    }, lastCount === "all" ? stringFormatter.format("selectedAll") : stringFormatter.format("selected", {
        count: lastCount
    })))));
}
const $7b5467653feb1153$var$ActionBarInnerWithRef = /*#__PURE__*/ (0, ($parcel$interopDefault($grxqE$react))).forwardRef($7b5467653feb1153$var$ActionBarInner);
/**
 * Action bars are used for single and bulk selection patterns when a user needs to perform actions on one or more items at the same time.
 */ const $7b5467653feb1153$export$e213cebad6250b4a = /*#__PURE__*/ (0, ($parcel$interopDefault($grxqE$react))).forwardRef($7b5467653feb1153$var$ActionBar);


/*
 * Copyright 2020 Adobe. All rights reserved.
 * This file is licensed to you under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License. You may obtain a copy
 * of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under
 * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 * OF ANY KIND, either express or implied. See the License for the specific language
 * governing permissions and limitations under the License.
 */ 




function $961b22d2b8bd2da1$var$ActionBarContainer(props, ref) {
    // Grabs specific props from the closest Provider (see https://react-spectrum.adobe.com/react-spectrum/Provider.html#property-groups). Remove if your component doesn't support any of the listed props.
    props = (0, $grxqE$reactspectrumprovider.useProviderProps)(props);
    let { children: children  } = props;
    let { styleProps: styleProps  } = (0, $grxqE$reactspectrumutils.useStyleProps)(props);
    let domRef = (0, $grxqE$reactspectrumutils.useDOMRef)(ref);
    return /*#__PURE__*/ (0, ($parcel$interopDefault($grxqE$react))).createElement("div", {
        ...(0, $grxqE$reactariautils.filterDOMProps)(props),
        ...styleProps,
        ref: domRef,
        className: (0, $grxqE$reactspectrumutils.classNames)((0, (/*@__PURE__*/$parcel$interopDefault($ff24c7bbe0c168db$exports))), "ActionBarContainer", styleProps.className)
    }, children);
}
/**
 * ActionBarContainer wraps around an ActionBar and a component that supports selection. It handles
 * the ActionBar's position with respect to its linked component.
 */ const $961b22d2b8bd2da1$export$ac2eb07f267e434c = /*#__PURE__*/ (0, ($parcel$interopDefault($grxqE$react))).forwardRef($961b22d2b8bd2da1$var$ActionBarContainer);
