// @react-spectrum/inlinealert/dist/import.mjs
import "./main.css";
import $d32NX$spectrumiconsuiAlertMedium from "@spectrum-icons/ui/AlertMedium";
import {useStyleProps as $d32NX$useStyleProps, useDOMRef as $d32NX$useDOMRef, classNames as $d32NX$classNames, SlotProvider as $d32NX$SlotProvider} from "@react-spectrum/utils";
import {filterDOMProps as $d32NX$filterDOMProps} from "@react-aria/utils";
import {Grid as $d32NX$Grid} from "@react-spectrum/layout";
import $d32NX$spectrumiconsuiInfoMedium from "@spectrum-icons/ui/InfoMedium";
import $d32NX$react from "react";
import $d32NX$spectrumiconsuiSuccessMedium from "@spectrum-icons/ui/SuccessMedium";
import {useLocalizedStringFormatter as $d32NX$useLocalizedStringFormatter} from "@react-aria/i18n";
import {useProviderProps as $d32NX$useProviderProps} from "@react-spectrum/provider";

function $parcel$interopDefault(a) {
  return a && a.__esModule ? a.default : a;
}
function $parcel$export(e, n, v, s) {
  Object.defineProperty(e, n, {get: v, set: s, enumerable: true, configurable: true});
}
/*
 * Copyright 2023 Adobe. All rights reserved.
 * This file is licensed to you under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License. You may obtain a copy
 * of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under
 * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 * OF ANY KIND, either express or implied. See the License for the specific language
 * governing permissions and limitations under the License.
 */ /// <reference types="css-module-types" />
/*
 * Copyright 2023 Adobe. All rights reserved.
 * This file is licensed to you under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License. You may obtain a copy
 * of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under
 * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 * OF ANY KIND, either express or implied. See the License for the specific language
 * governing permissions and limitations under the License.
 */ 




var $a0360622247416a9$exports = {};
var $7c06d136a7db332f$exports = {};
$7c06d136a7db332f$exports = {
    "info": `معلومات`,
    "negative": `خطأ`,
    "notice": `تحذير`,
    "positive": `تم بنجاح`
};


var $7572fcde1b201200$exports = {};
$7572fcde1b201200$exports = {
    "info": `Информация`,
    "negative": `Грешка`,
    "notice": `Предупреждение`,
    "positive": `Успешно`
};


var $4baaa05e7e36c8d5$exports = {};
$4baaa05e7e36c8d5$exports = {
    "info": `Informace`,
    "negative": `Chyba`,
    "notice": `Varování`,
    "positive": `Úspěch`
};


var $67391f9ec3919bbd$exports = {};
$67391f9ec3919bbd$exports = {
    "info": `Oplysninger`,
    "negative": `Fejl`,
    "notice": `Advarsel`,
    "positive": `Fuldført`
};


var $ba4d55b7a25dbd70$exports = {};
$ba4d55b7a25dbd70$exports = {
    "info": `Informationen`,
    "negative": `Fehler`,
    "notice": `Warnung`,
    "positive": `Erfolg`
};


var $8f11cbfb87576513$exports = {};
$8f11cbfb87576513$exports = {
    "info": `Πληροφορίες`,
    "negative": `Σφάλμα`,
    "notice": `Προειδοποίηση`,
    "positive": `Επιτυχία`
};


var $a5983f3344cffe9c$exports = {};
$a5983f3344cffe9c$exports = {
    "negative": `Error`,
    "notice": `Warning`,
    "info": `Information`,
    "positive": `Success`
};


var $5b00f4541c9c46ff$exports = {};
$5b00f4541c9c46ff$exports = {
    "info": `Información`,
    "negative": `Error`,
    "notice": `Advertencia`,
    "positive": `Éxito`
};


var $aa360d77330765aa$exports = {};
$aa360d77330765aa$exports = {
    "info": `Teave`,
    "negative": `Tõrge`,
    "notice": `Hoiatus`,
    "positive": `Valmis`
};


var $03e931e04f228c9f$exports = {};
$03e931e04f228c9f$exports = {
    "info": `Tiedot`,
    "negative": `Virhe`,
    "notice": `Varoitus`,
    "positive": `Onnistui`
};


var $9d5074a25853ca6e$exports = {};
$9d5074a25853ca6e$exports = {
    "info": `Informations`,
    "negative": `Erreur`,
    "notice": `Avertissement`,
    "positive": `Succès`
};


var $2c2676389f361ece$exports = {};
$2c2676389f361ece$exports = {
    "info": `מידע`,
    "negative": `שגיאה`,
    "notice": `אזהרה`,
    "positive": `הצלחה`
};


var $5fdda27174ef2fc7$exports = {};
$5fdda27174ef2fc7$exports = {
    "info": `Informacije`,
    "negative": `Pogreška`,
    "notice": `Upozorenje`,
    "positive": `Uspješno`
};


var $dc458e3abea04ded$exports = {};
$dc458e3abea04ded$exports = {
    "info": `Információ`,
    "negative": `Hiba`,
    "notice": `Figyelmeztetés`,
    "positive": `Siker`
};


var $2ebc0e35fdc040b7$exports = {};
$2ebc0e35fdc040b7$exports = {
    "info": `Informazioni`,
    "negative": `Errore`,
    "notice": `Avviso`,
    "positive": `Operazione riuscita`
};


var $44f02920600efced$exports = {};
$44f02920600efced$exports = {
    "info": `情報`,
    "negative": `エラー`,
    "notice": `警告`,
    "positive": `成功`
};


var $eeee9839fa747b1b$exports = {};
$eeee9839fa747b1b$exports = {
    "info": `정보`,
    "negative": `오류`,
    "notice": `경고`,
    "positive": `성공`
};


var $7889661aab511c26$exports = {};
$7889661aab511c26$exports = {
    "info": `Informacija`,
    "negative": `Klaida`,
    "notice": `Įspėjimas`,
    "positive": `Sėkmingai`
};


var $f73d040d5b290afc$exports = {};
$f73d040d5b290afc$exports = {
    "info": `Informācija`,
    "negative": `Kļūda`,
    "notice": `Brīdinājums`,
    "positive": `Izdevās`
};


var $eae0cb6abb6f7592$exports = {};
$eae0cb6abb6f7592$exports = {
    "info": `Informasjon`,
    "negative": `Feil`,
    "notice": `Advarsel`,
    "positive": `Vellykket`
};


var $1e0495a2c0782298$exports = {};
$1e0495a2c0782298$exports = {
    "info": `Informatie`,
    "negative": `Fout`,
    "notice": `Waarschuwing`,
    "positive": `Geslaagd`
};


var $06b952e4a8b5ac61$exports = {};
$06b952e4a8b5ac61$exports = {
    "info": `Informacja`,
    "negative": `Błąd`,
    "notice": `Ostrzeżenie`,
    "positive": `Powodzenie`
};


var $d6a05b7bb1801915$exports = {};
$d6a05b7bb1801915$exports = {
    "info": `Informações`,
    "negative": `Erro`,
    "notice": `Aviso`,
    "positive": `Sucesso`
};


var $83d519ea719f843b$exports = {};
$83d519ea719f843b$exports = {
    "info": `Informação`,
    "negative": `Erro`,
    "notice": `Aviso`,
    "positive": `Sucesso`
};


var $3fde5b55008eef3f$exports = {};
$3fde5b55008eef3f$exports = {
    "info": `Informații`,
    "negative": `Eroare`,
    "notice": `Avertisment`,
    "positive": `Succes`
};


var $96d433eefb3f8906$exports = {};
$96d433eefb3f8906$exports = {
    "info": `Информация`,
    "negative": `Ошибка`,
    "notice": `Предупреждение`,
    "positive": `Успешно`
};


var $0893538688618f33$exports = {};
$0893538688618f33$exports = {
    "info": `Informácie`,
    "negative": `Chyba`,
    "notice": `Upozornenie`,
    "positive": `Úspech`
};


var $254b6c961e4f0e80$exports = {};
$254b6c961e4f0e80$exports = {
    "info": `Informacije`,
    "negative": `Napaka`,
    "notice": `Opozorilo`,
    "positive": `Uspešno`
};


var $680d309814a11fc5$exports = {};
$680d309814a11fc5$exports = {
    "info": `Informacije`,
    "negative": `Greška`,
    "notice": `Upozorenje`,
    "positive": `Uspešno`
};


var $f9418595c1e772cf$exports = {};
$f9418595c1e772cf$exports = {
    "info": `Information`,
    "negative": `Fel`,
    "notice": `Varning`,
    "positive": `Lyckades`
};


var $f7b3644c5b6100d7$exports = {};
$f7b3644c5b6100d7$exports = {
    "info": `Bilgiler`,
    "negative": `Hata`,
    "notice": `Uyarı`,
    "positive": `Başarılı`
};


var $74a01294f8d465f8$exports = {};
$74a01294f8d465f8$exports = {
    "info": `Інформація`,
    "negative": `Помилка`,
    "notice": `Попередження`,
    "positive": `Успішно`
};


var $a1438ca118687381$exports = {};
$a1438ca118687381$exports = {
    "info": `信息`,
    "negative": `错误`,
    "notice": `警告`,
    "positive": `成功`
};


var $948f0650714aa894$exports = {};
$948f0650714aa894$exports = {
    "info": `資訊`,
    "negative": `錯誤`,
    "notice": `警告`,
    "positive": `成功`
};


$a0360622247416a9$exports = {
    "ar-AE": $7c06d136a7db332f$exports,
    "bg-BG": $7572fcde1b201200$exports,
    "cs-CZ": $4baaa05e7e36c8d5$exports,
    "da-DK": $67391f9ec3919bbd$exports,
    "de-DE": $ba4d55b7a25dbd70$exports,
    "el-GR": $8f11cbfb87576513$exports,
    "en-US": $a5983f3344cffe9c$exports,
    "es-ES": $5b00f4541c9c46ff$exports,
    "et-EE": $aa360d77330765aa$exports,
    "fi-FI": $03e931e04f228c9f$exports,
    "fr-FR": $9d5074a25853ca6e$exports,
    "he-IL": $2c2676389f361ece$exports,
    "hr-HR": $5fdda27174ef2fc7$exports,
    "hu-HU": $dc458e3abea04ded$exports,
    "it-IT": $2ebc0e35fdc040b7$exports,
    "ja-JP": $44f02920600efced$exports,
    "ko-KR": $eeee9839fa747b1b$exports,
    "lt-LT": $7889661aab511c26$exports,
    "lv-LV": $f73d040d5b290afc$exports,
    "nb-NO": $eae0cb6abb6f7592$exports,
    "nl-NL": $1e0495a2c0782298$exports,
    "pl-PL": $06b952e4a8b5ac61$exports,
    "pt-BR": $d6a05b7bb1801915$exports,
    "pt-PT": $83d519ea719f843b$exports,
    "ro-RO": $3fde5b55008eef3f$exports,
    "ru-RU": $96d433eefb3f8906$exports,
    "sk-SK": $0893538688618f33$exports,
    "sl-SI": $254b6c961e4f0e80$exports,
    "sr-SP": $680d309814a11fc5$exports,
    "sv-SE": $f9418595c1e772cf$exports,
    "tr-TR": $f7b3644c5b6100d7$exports,
    "uk-UA": $74a01294f8d465f8$exports,
    "zh-CN": $a1438ca118687381$exports,
    "zh-TW": $948f0650714aa894$exports
};



var $48fe1f4c438508c0$exports = {};

$parcel$export($48fe1f4c438508c0$exports, "spectrum-InLineAlert", () => $48fe1f4c438508c0$export$73769bb5d26d9ff7, (v) => $48fe1f4c438508c0$export$73769bb5d26d9ff7 = v);
$parcel$export($48fe1f4c438508c0$exports, "spectrum-InLineAlert-grid", () => $48fe1f4c438508c0$export$c31ea092819cec48, (v) => $48fe1f4c438508c0$export$c31ea092819cec48 = v);
$parcel$export($48fe1f4c438508c0$exports, "spectrum-InLineAlert--neutral", () => $48fe1f4c438508c0$export$1d6dd31283ff6c5c, (v) => $48fe1f4c438508c0$export$1d6dd31283ff6c5c = v);
$parcel$export($48fe1f4c438508c0$exports, "spectrum-InLineAlert-icon", () => $48fe1f4c438508c0$export$3be02ea40f92bdfc, (v) => $48fe1f4c438508c0$export$3be02ea40f92bdfc = v);
$parcel$export($48fe1f4c438508c0$exports, "spectrum-InLineAlert-heading", () => $48fe1f4c438508c0$export$5f11e05135494154, (v) => $48fe1f4c438508c0$export$5f11e05135494154 = v);
$parcel$export($48fe1f4c438508c0$exports, "spectrum-InLineAlert-content", () => $48fe1f4c438508c0$export$b9bc26a202328584, (v) => $48fe1f4c438508c0$export$b9bc26a202328584 = v);
$parcel$export($48fe1f4c438508c0$exports, "spectrum-InLineAlert--info", () => $48fe1f4c438508c0$export$71c1df0099076ffd, (v) => $48fe1f4c438508c0$export$71c1df0099076ffd = v);
$parcel$export($48fe1f4c438508c0$exports, "spectrum-InLineAlert--notice", () => $48fe1f4c438508c0$export$29b25aea79cc7183, (v) => $48fe1f4c438508c0$export$29b25aea79cc7183 = v);
$parcel$export($48fe1f4c438508c0$exports, "spectrum-InLineAlert--positive", () => $48fe1f4c438508c0$export$fe577b621f09240b, (v) => $48fe1f4c438508c0$export$fe577b621f09240b = v);
$parcel$export($48fe1f4c438508c0$exports, "spectrum-InLineAlert--negative", () => $48fe1f4c438508c0$export$22162c465dd1f294, (v) => $48fe1f4c438508c0$export$22162c465dd1f294 = v);
var $48fe1f4c438508c0$export$73769bb5d26d9ff7;
var $48fe1f4c438508c0$export$c31ea092819cec48;
var $48fe1f4c438508c0$export$1d6dd31283ff6c5c;
var $48fe1f4c438508c0$export$3be02ea40f92bdfc;
var $48fe1f4c438508c0$export$5f11e05135494154;
var $48fe1f4c438508c0$export$b9bc26a202328584;
var $48fe1f4c438508c0$export$71c1df0099076ffd;
var $48fe1f4c438508c0$export$29b25aea79cc7183;
var $48fe1f4c438508c0$export$fe577b621f09240b;
var $48fe1f4c438508c0$export$22162c465dd1f294;
$48fe1f4c438508c0$export$73769bb5d26d9ff7 = "spectrum-InLineAlert_479995";
$48fe1f4c438508c0$export$c31ea092819cec48 = "spectrum-InLineAlert-grid_479995";
$48fe1f4c438508c0$export$1d6dd31283ff6c5c = "spectrum-InLineAlert--neutral_479995";
$48fe1f4c438508c0$export$3be02ea40f92bdfc = "spectrum-InLineAlert-icon_479995";
$48fe1f4c438508c0$export$5f11e05135494154 = "spectrum-InLineAlert-heading_479995";
$48fe1f4c438508c0$export$b9bc26a202328584 = "spectrum-InLineAlert-content_479995";
$48fe1f4c438508c0$export$71c1df0099076ffd = "spectrum-InLineAlert--info_479995";
$48fe1f4c438508c0$export$29b25aea79cc7183 = "spectrum-InLineAlert--notice_479995";
$48fe1f4c438508c0$export$fe577b621f09240b = "spectrum-InLineAlert--positive_479995";
$48fe1f4c438508c0$export$22162c465dd1f294 = "spectrum-InLineAlert--negative_479995";





let $abf1e6290496dda5$var$ICONS = {
    info: (0, $d32NX$spectrumiconsuiInfoMedium),
    positive: (0, $d32NX$spectrumiconsuiSuccessMedium),
    notice: (0, $d32NX$spectrumiconsuiAlertMedium),
    negative: (0, $d32NX$spectrumiconsuiAlertMedium)
};
function $abf1e6290496dda5$var$InlineAlert(props, ref) {
    props = (0, $d32NX$useProviderProps)(props);
    let { children: children , variant: variant = "neutral" , ...otherProps } = props;
    let { styleProps: styleProps  } = (0, $d32NX$useStyleProps)(otherProps);
    let domRef = (0, $d32NX$useDOMRef)(ref);
    let slots = {
        heading: {
            UNSAFE_className: (0, (/*@__PURE__*/$parcel$interopDefault($48fe1f4c438508c0$exports)))["spectrum-InLineAlert-heading"]
        },
        content: {
            UNSAFE_className: (0, (/*@__PURE__*/$parcel$interopDefault($48fe1f4c438508c0$exports)))["spectrum-InLineAlert-content"]
        }
    };
    let stringFormatter = (0, $d32NX$useLocalizedStringFormatter)((0, (/*@__PURE__*/$parcel$interopDefault($a0360622247416a9$exports))));
    let Icon = null;
    let iconAlt;
    if (variant in $abf1e6290496dda5$var$ICONS) {
        Icon = $abf1e6290496dda5$var$ICONS[variant];
        iconAlt = stringFormatter.format(variant);
    }
    return /*#__PURE__*/ (0, $d32NX$react).createElement("div", {
        ...(0, $d32NX$filterDOMProps)(props),
        ...styleProps,
        ref: domRef,
        className: (0, $d32NX$classNames)((0, (/*@__PURE__*/$parcel$interopDefault($48fe1f4c438508c0$exports))), "spectrum-InLineAlert", `spectrum-InLineAlert--${variant}`, styleProps.className),
        role: "alert"
    }, /*#__PURE__*/ (0, $d32NX$react).createElement((0, $d32NX$Grid), {
        UNSAFE_className: (0, (/*@__PURE__*/$parcel$interopDefault($48fe1f4c438508c0$exports)))["spectrum-InLineAlert-grid"]
    }, /*#__PURE__*/ (0, $d32NX$react).createElement((0, $d32NX$SlotProvider), {
        slots: slots
    }, Icon && /*#__PURE__*/ (0, $d32NX$react).createElement(Icon, {
        UNSAFE_className: (0, (/*@__PURE__*/$parcel$interopDefault($48fe1f4c438508c0$exports)))["spectrum-InLineAlert-icon"],
        "aria-label": iconAlt
    }), children)));
}
/**
 * Inline alerts display a non-modal message associated with objects in a view.
 * These are often used in form validation, providing a place to aggregate feedback related to multiple fields.
 */ const $abf1e6290496dda5$export$a3b2c96db9b0eb71 = /*#__PURE__*/ (0, $d32NX$react).forwardRef($abf1e6290496dda5$var$InlineAlert);




export {$abf1e6290496dda5$export$a3b2c96db9b0eb71 as InlineAlert};
