// @react-spectrum/inlinealert/dist/main.js
require("./main.css");
var $RX49b$spectrumiconsuiAlertMedium = require("@spectrum-icons/ui/AlertMedium");
var $RX49b$reactspectrumutils = require("@react-spectrum/utils");
var $RX49b$reactariautils = require("@react-aria/utils");
var $RX49b$reactspectrumlayout = require("@react-spectrum/layout");
var $RX49b$spectrumiconsuiInfoMedium = require("@spectrum-icons/ui/InfoMedium");
var $RX49b$react = require("react");
var $RX49b$spectrumiconsuiSuccessMedium = require("@spectrum-icons/ui/SuccessMedium");
var $RX49b$reactariai18n = require("@react-aria/i18n");
var $RX49b$reactspectrumprovider = require("@react-spectrum/provider");

function $parcel$export(e, n, v, s) {
  Object.defineProperty(e, n, {get: v, set: s, enumerable: true, configurable: true});
}
function $parcel$interopDefault(a) {
  return a && a.__esModule ? a.default : a;
}

$parcel$export(module.exports, "InlineAlert", () => $5de8c6729f05c665$export$a3b2c96db9b0eb71);
/*
 * Copyright 2023 Adobe. All rights reserved.
 * This file is licensed to you under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License. You may obtain a copy
 * of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under
 * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 * OF ANY KIND, either express or implied. See the License for the specific language
 * governing permissions and limitations under the License.
 */ /// <reference types="css-module-types" />
/*
 * Copyright 2023 Adobe. All rights reserved.
 * This file is licensed to you under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License. You may obtain a copy
 * of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under
 * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 * OF ANY KIND, either express or implied. See the License for the specific language
 * governing permissions and limitations under the License.
 */ 




var $cf1d087b199fc7c5$exports = {};
var $70305e81a8d2187b$exports = {};
$70305e81a8d2187b$exports = {
    "info": `معلومات`,
    "negative": `خطأ`,
    "notice": `تحذير`,
    "positive": `تم بنجاح`
};


var $0c660aaf1ba719ea$exports = {};
$0c660aaf1ba719ea$exports = {
    "info": `Информация`,
    "negative": `Грешка`,
    "notice": `Предупреждение`,
    "positive": `Успешно`
};


var $09ed32d40da22228$exports = {};
$09ed32d40da22228$exports = {
    "info": `Informace`,
    "negative": `Chyba`,
    "notice": `Varování`,
    "positive": `Úspěch`
};


var $fb9891466d18e5a4$exports = {};
$fb9891466d18e5a4$exports = {
    "info": `Oplysninger`,
    "negative": `Fejl`,
    "notice": `Advarsel`,
    "positive": `Fuldført`
};


var $5ccd3660efbdf1ba$exports = {};
$5ccd3660efbdf1ba$exports = {
    "info": `Informationen`,
    "negative": `Fehler`,
    "notice": `Warnung`,
    "positive": `Erfolg`
};


var $d99fc87568761416$exports = {};
$d99fc87568761416$exports = {
    "info": `Πληροφορίες`,
    "negative": `Σφάλμα`,
    "notice": `Προειδοποίηση`,
    "positive": `Επιτυχία`
};


var $fc814fd646f97b4a$exports = {};
$fc814fd646f97b4a$exports = {
    "negative": `Error`,
    "notice": `Warning`,
    "info": `Information`,
    "positive": `Success`
};


var $cd2deb9e4ace11a8$exports = {};
$cd2deb9e4ace11a8$exports = {
    "info": `Información`,
    "negative": `Error`,
    "notice": `Advertencia`,
    "positive": `Éxito`
};


var $f0d5c61765f22ed3$exports = {};
$f0d5c61765f22ed3$exports = {
    "info": `Teave`,
    "negative": `Tõrge`,
    "notice": `Hoiatus`,
    "positive": `Valmis`
};


var $a2bb977ca4ce5ca1$exports = {};
$a2bb977ca4ce5ca1$exports = {
    "info": `Tiedot`,
    "negative": `Virhe`,
    "notice": `Varoitus`,
    "positive": `Onnistui`
};


var $116c45d32f2f45ca$exports = {};
$116c45d32f2f45ca$exports = {
    "info": `Informations`,
    "negative": `Erreur`,
    "notice": `Avertissement`,
    "positive": `Succès`
};


var $1071566f78693587$exports = {};
$1071566f78693587$exports = {
    "info": `מידע`,
    "negative": `שגיאה`,
    "notice": `אזהרה`,
    "positive": `הצלחה`
};


var $c636cbe8751599df$exports = {};
$c636cbe8751599df$exports = {
    "info": `Informacije`,
    "negative": `Pogreška`,
    "notice": `Upozorenje`,
    "positive": `Uspješno`
};


var $2f9ee8544d50270a$exports = {};
$2f9ee8544d50270a$exports = {
    "info": `Információ`,
    "negative": `Hiba`,
    "notice": `Figyelmeztetés`,
    "positive": `Siker`
};


var $c3a100d558fcd0c3$exports = {};
$c3a100d558fcd0c3$exports = {
    "info": `Informazioni`,
    "negative": `Errore`,
    "notice": `Avviso`,
    "positive": `Operazione riuscita`
};


var $9b6747fa80c2005c$exports = {};
$9b6747fa80c2005c$exports = {
    "info": `情報`,
    "negative": `エラー`,
    "notice": `警告`,
    "positive": `成功`
};


var $7aa436d0bc11d8df$exports = {};
$7aa436d0bc11d8df$exports = {
    "info": `정보`,
    "negative": `오류`,
    "notice": `경고`,
    "positive": `성공`
};


var $2306a8fe8a217aae$exports = {};
$2306a8fe8a217aae$exports = {
    "info": `Informacija`,
    "negative": `Klaida`,
    "notice": `Įspėjimas`,
    "positive": `Sėkmingai`
};


var $7ee88da038778f20$exports = {};
$7ee88da038778f20$exports = {
    "info": `Informācija`,
    "negative": `Kļūda`,
    "notice": `Brīdinājums`,
    "positive": `Izdevās`
};


var $4ba767217b9690c2$exports = {};
$4ba767217b9690c2$exports = {
    "info": `Informasjon`,
    "negative": `Feil`,
    "notice": `Advarsel`,
    "positive": `Vellykket`
};


var $3c4e9f1e46e8db66$exports = {};
$3c4e9f1e46e8db66$exports = {
    "info": `Informatie`,
    "negative": `Fout`,
    "notice": `Waarschuwing`,
    "positive": `Geslaagd`
};


var $15ba96b5246db651$exports = {};
$15ba96b5246db651$exports = {
    "info": `Informacja`,
    "negative": `Błąd`,
    "notice": `Ostrzeżenie`,
    "positive": `Powodzenie`
};


var $969881ee58ec3452$exports = {};
$969881ee58ec3452$exports = {
    "info": `Informações`,
    "negative": `Erro`,
    "notice": `Aviso`,
    "positive": `Sucesso`
};


var $ded9cae395371ad3$exports = {};
$ded9cae395371ad3$exports = {
    "info": `Informação`,
    "negative": `Erro`,
    "notice": `Aviso`,
    "positive": `Sucesso`
};


var $9966420945e42ed8$exports = {};
$9966420945e42ed8$exports = {
    "info": `Informații`,
    "negative": `Eroare`,
    "notice": `Avertisment`,
    "positive": `Succes`
};


var $9c9f39649e3598f2$exports = {};
$9c9f39649e3598f2$exports = {
    "info": `Информация`,
    "negative": `Ошибка`,
    "notice": `Предупреждение`,
    "positive": `Успешно`
};


var $d9dce3fa927805b2$exports = {};
$d9dce3fa927805b2$exports = {
    "info": `Informácie`,
    "negative": `Chyba`,
    "notice": `Upozornenie`,
    "positive": `Úspech`
};


var $bdb67023e40cc961$exports = {};
$bdb67023e40cc961$exports = {
    "info": `Informacije`,
    "negative": `Napaka`,
    "notice": `Opozorilo`,
    "positive": `Uspešno`
};


var $21eca3566d4fbc9e$exports = {};
$21eca3566d4fbc9e$exports = {
    "info": `Informacije`,
    "negative": `Greška`,
    "notice": `Upozorenje`,
    "positive": `Uspešno`
};


var $77c4ed2ec49f90d9$exports = {};
$77c4ed2ec49f90d9$exports = {
    "info": `Information`,
    "negative": `Fel`,
    "notice": `Varning`,
    "positive": `Lyckades`
};


var $aea4f814e594ee44$exports = {};
$aea4f814e594ee44$exports = {
    "info": `Bilgiler`,
    "negative": `Hata`,
    "notice": `Uyarı`,
    "positive": `Başarılı`
};


var $ed1255c500b98f56$exports = {};
$ed1255c500b98f56$exports = {
    "info": `Інформація`,
    "negative": `Помилка`,
    "notice": `Попередження`,
    "positive": `Успішно`
};


var $2adae4170a5c1992$exports = {};
$2adae4170a5c1992$exports = {
    "info": `信息`,
    "negative": `错误`,
    "notice": `警告`,
    "positive": `成功`
};


var $7cf811ce2d1c9d87$exports = {};
$7cf811ce2d1c9d87$exports = {
    "info": `資訊`,
    "negative": `錯誤`,
    "notice": `警告`,
    "positive": `成功`
};


$cf1d087b199fc7c5$exports = {
    "ar-AE": $70305e81a8d2187b$exports,
    "bg-BG": $0c660aaf1ba719ea$exports,
    "cs-CZ": $09ed32d40da22228$exports,
    "da-DK": $fb9891466d18e5a4$exports,
    "de-DE": $5ccd3660efbdf1ba$exports,
    "el-GR": $d99fc87568761416$exports,
    "en-US": $fc814fd646f97b4a$exports,
    "es-ES": $cd2deb9e4ace11a8$exports,
    "et-EE": $f0d5c61765f22ed3$exports,
    "fi-FI": $a2bb977ca4ce5ca1$exports,
    "fr-FR": $116c45d32f2f45ca$exports,
    "he-IL": $1071566f78693587$exports,
    "hr-HR": $c636cbe8751599df$exports,
    "hu-HU": $2f9ee8544d50270a$exports,
    "it-IT": $c3a100d558fcd0c3$exports,
    "ja-JP": $9b6747fa80c2005c$exports,
    "ko-KR": $7aa436d0bc11d8df$exports,
    "lt-LT": $2306a8fe8a217aae$exports,
    "lv-LV": $7ee88da038778f20$exports,
    "nb-NO": $4ba767217b9690c2$exports,
    "nl-NL": $3c4e9f1e46e8db66$exports,
    "pl-PL": $15ba96b5246db651$exports,
    "pt-BR": $969881ee58ec3452$exports,
    "pt-PT": $ded9cae395371ad3$exports,
    "ro-RO": $9966420945e42ed8$exports,
    "ru-RU": $9c9f39649e3598f2$exports,
    "sk-SK": $d9dce3fa927805b2$exports,
    "sl-SI": $bdb67023e40cc961$exports,
    "sr-SP": $21eca3566d4fbc9e$exports,
    "sv-SE": $77c4ed2ec49f90d9$exports,
    "tr-TR": $aea4f814e594ee44$exports,
    "uk-UA": $ed1255c500b98f56$exports,
    "zh-CN": $2adae4170a5c1992$exports,
    "zh-TW": $7cf811ce2d1c9d87$exports
};



var $8d976ce4ebad141b$exports = {};

$parcel$export($8d976ce4ebad141b$exports, "spectrum-InLineAlert", () => $8d976ce4ebad141b$export$73769bb5d26d9ff7, (v) => $8d976ce4ebad141b$export$73769bb5d26d9ff7 = v);
$parcel$export($8d976ce4ebad141b$exports, "spectrum-InLineAlert-grid", () => $8d976ce4ebad141b$export$c31ea092819cec48, (v) => $8d976ce4ebad141b$export$c31ea092819cec48 = v);
$parcel$export($8d976ce4ebad141b$exports, "spectrum-InLineAlert--neutral", () => $8d976ce4ebad141b$export$1d6dd31283ff6c5c, (v) => $8d976ce4ebad141b$export$1d6dd31283ff6c5c = v);
$parcel$export($8d976ce4ebad141b$exports, "spectrum-InLineAlert-icon", () => $8d976ce4ebad141b$export$3be02ea40f92bdfc, (v) => $8d976ce4ebad141b$export$3be02ea40f92bdfc = v);
$parcel$export($8d976ce4ebad141b$exports, "spectrum-InLineAlert-heading", () => $8d976ce4ebad141b$export$5f11e05135494154, (v) => $8d976ce4ebad141b$export$5f11e05135494154 = v);
$parcel$export($8d976ce4ebad141b$exports, "spectrum-InLineAlert-content", () => $8d976ce4ebad141b$export$b9bc26a202328584, (v) => $8d976ce4ebad141b$export$b9bc26a202328584 = v);
$parcel$export($8d976ce4ebad141b$exports, "spectrum-InLineAlert--info", () => $8d976ce4ebad141b$export$71c1df0099076ffd, (v) => $8d976ce4ebad141b$export$71c1df0099076ffd = v);
$parcel$export($8d976ce4ebad141b$exports, "spectrum-InLineAlert--notice", () => $8d976ce4ebad141b$export$29b25aea79cc7183, (v) => $8d976ce4ebad141b$export$29b25aea79cc7183 = v);
$parcel$export($8d976ce4ebad141b$exports, "spectrum-InLineAlert--positive", () => $8d976ce4ebad141b$export$fe577b621f09240b, (v) => $8d976ce4ebad141b$export$fe577b621f09240b = v);
$parcel$export($8d976ce4ebad141b$exports, "spectrum-InLineAlert--negative", () => $8d976ce4ebad141b$export$22162c465dd1f294, (v) => $8d976ce4ebad141b$export$22162c465dd1f294 = v);
var $8d976ce4ebad141b$export$73769bb5d26d9ff7;
var $8d976ce4ebad141b$export$c31ea092819cec48;
var $8d976ce4ebad141b$export$1d6dd31283ff6c5c;
var $8d976ce4ebad141b$export$3be02ea40f92bdfc;
var $8d976ce4ebad141b$export$5f11e05135494154;
var $8d976ce4ebad141b$export$b9bc26a202328584;
var $8d976ce4ebad141b$export$71c1df0099076ffd;
var $8d976ce4ebad141b$export$29b25aea79cc7183;
var $8d976ce4ebad141b$export$fe577b621f09240b;
var $8d976ce4ebad141b$export$22162c465dd1f294;
$8d976ce4ebad141b$export$73769bb5d26d9ff7 = "spectrum-InLineAlert_479995";
$8d976ce4ebad141b$export$c31ea092819cec48 = "spectrum-InLineAlert-grid_479995";
$8d976ce4ebad141b$export$1d6dd31283ff6c5c = "spectrum-InLineAlert--neutral_479995";
$8d976ce4ebad141b$export$3be02ea40f92bdfc = "spectrum-InLineAlert-icon_479995";
$8d976ce4ebad141b$export$5f11e05135494154 = "spectrum-InLineAlert-heading_479995";
$8d976ce4ebad141b$export$b9bc26a202328584 = "spectrum-InLineAlert-content_479995";
$8d976ce4ebad141b$export$71c1df0099076ffd = "spectrum-InLineAlert--info_479995";
$8d976ce4ebad141b$export$29b25aea79cc7183 = "spectrum-InLineAlert--notice_479995";
$8d976ce4ebad141b$export$fe577b621f09240b = "spectrum-InLineAlert--positive_479995";
$8d976ce4ebad141b$export$22162c465dd1f294 = "spectrum-InLineAlert--negative_479995";





let $5de8c6729f05c665$var$ICONS = {
    info: (0, ($parcel$interopDefault($RX49b$spectrumiconsuiInfoMedium))),
    positive: (0, ($parcel$interopDefault($RX49b$spectrumiconsuiSuccessMedium))),
    notice: (0, ($parcel$interopDefault($RX49b$spectrumiconsuiAlertMedium))),
    negative: (0, ($parcel$interopDefault($RX49b$spectrumiconsuiAlertMedium)))
};
function $5de8c6729f05c665$var$InlineAlert(props, ref) {
    props = (0, $RX49b$reactspectrumprovider.useProviderProps)(props);
    let { children: children , variant: variant = "neutral" , ...otherProps } = props;
    let { styleProps: styleProps  } = (0, $RX49b$reactspectrumutils.useStyleProps)(otherProps);
    let domRef = (0, $RX49b$reactspectrumutils.useDOMRef)(ref);
    let slots = {
        heading: {
            UNSAFE_className: (0, (/*@__PURE__*/$parcel$interopDefault($8d976ce4ebad141b$exports)))["spectrum-InLineAlert-heading"]
        },
        content: {
            UNSAFE_className: (0, (/*@__PURE__*/$parcel$interopDefault($8d976ce4ebad141b$exports)))["spectrum-InLineAlert-content"]
        }
    };
    let stringFormatter = (0, $RX49b$reactariai18n.useLocalizedStringFormatter)((0, (/*@__PURE__*/$parcel$interopDefault($cf1d087b199fc7c5$exports))));
    let Icon = null;
    let iconAlt;
    if (variant in $5de8c6729f05c665$var$ICONS) {
        Icon = $5de8c6729f05c665$var$ICONS[variant];
        iconAlt = stringFormatter.format(variant);
    }
    return /*#__PURE__*/ (0, ($parcel$interopDefault($RX49b$react))).createElement("div", {
        ...(0, $RX49b$reactariautils.filterDOMProps)(props),
        ...styleProps,
        ref: domRef,
        className: (0, $RX49b$reactspectrumutils.classNames)((0, (/*@__PURE__*/$parcel$interopDefault($8d976ce4ebad141b$exports))), "spectrum-InLineAlert", `spectrum-InLineAlert--${variant}`, styleProps.className),
        role: "alert"
    }, /*#__PURE__*/ (0, ($parcel$interopDefault($RX49b$react))).createElement((0, $RX49b$reactspectrumlayout.Grid), {
        UNSAFE_className: (0, (/*@__PURE__*/$parcel$interopDefault($8d976ce4ebad141b$exports)))["spectrum-InLineAlert-grid"]
    }, /*#__PURE__*/ (0, ($parcel$interopDefault($RX49b$react))).createElement((0, $RX49b$reactspectrumutils.SlotProvider), {
        slots: slots
    }, Icon && /*#__PURE__*/ (0, ($parcel$interopDefault($RX49b$react))).createElement(Icon, {
        UNSAFE_className: (0, (/*@__PURE__*/$parcel$interopDefault($8d976ce4ebad141b$exports)))["spectrum-InLineAlert-icon"],
        "aria-label": iconAlt
    }), children)));
}
/**
 * Inline alerts display a non-modal message associated with objects in a view.
 * These are often used in form validation, providing a place to aggregate feedback related to multiple fields.
 */ const $5de8c6729f05c665$export$a3b2c96db9b0eb71 = /*#__PURE__*/ (0, ($parcel$interopDefault($RX49b$react))).forwardRef($5de8c6729f05c665$var$InlineAlert);
