// @react-spectrum/menu/dist/main.js
require("./main.css");
var $3ThwF$reactstatelycollections = require("@react-stately/collections");
var $3ThwF$reactspectrumutils = require("@react-spectrum/utils");
var $3ThwF$reactspectrumoverlays = require("@react-spectrum/overlays");
var $3ThwF$reactariainteractions = require("@react-aria/interactions");
var $3ThwF$react = require("react");
var $3ThwF$reactariamenu = require("@react-aria/menu");
var $3ThwF$reactstatelymenu = require("@react-stately/menu");
var $3ThwF$reactariafocus = require("@react-aria/focus");
var $3ThwF$reactariautils = require("@react-aria/utils");
var $3ThwF$reactstatelytree = require("@react-stately/tree");
var $3ThwF$spectrumiconsuiCheckmarkMedium = require("@spectrum-icons/ui/CheckmarkMedium");
var $3ThwF$reactspectrumlayout = require("@react-spectrum/layout");
var $3ThwF$spectrumiconsworkflowInfoOutline = require("@spectrum-icons/workflow/InfoOutline");
var $3ThwF$reactspectrumtext = require("@react-spectrum/text");
var $3ThwF$reactariai18n = require("@react-aria/i18n");
var $3ThwF$reactariaseparator = require("@react-aria/separator");
var $3ThwF$reactspectrumbutton = require("@react-spectrum/button");
var $3ThwF$spectrumiconsworkflowMore = require("@spectrum-icons/workflow/More");
var $3ThwF$reactariaoverlays = require("@react-aria/overlays");
var $3ThwF$reactstatelyoverlays = require("@react-stately/overlays");

function $parcel$export(e, n, v, s) {
  Object.defineProperty(e, n, {get: v, set: s, enumerable: true, configurable: true});
}
function $parcel$interopDefault(a) {
  return a && a.__esModule ? a.default : a;
}

$parcel$export(module.exports, "MenuTrigger", () => $e56311f7204399c7$export$27d2ad3c5815583e);
$parcel$export(module.exports, "Menu", () => $63a7dff9cbe2d046$export$d9b273488cd8ce6f);
$parcel$export(module.exports, "ActionMenu", () => $53a2f2636c97a561$export$ed57a210b9d13bb6);
$parcel$export(module.exports, "ContextualHelpTrigger", () => $db380858d398b658$export$5413b169fff83e61);
$parcel$export(module.exports, "Item", () => $3ThwF$reactstatelycollections.Item);
$parcel$export(module.exports, "Section", () => $3ThwF$reactstatelycollections.Section);
/*
 * Copyright 2020 Adobe. All rights reserved.
 * This file is licensed to you under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License. You may obtain a copy
 * of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under
 * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 * OF ANY KIND, either express or implied. See the License for the specific language
 * governing permissions and limitations under the License.
 */ /// <reference types="css-module-types" />
/*
 * Copyright 2020 Adobe. All rights reserved.
 * This file is licensed to you under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License. You may obtain a copy
 * of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under
 * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 * OF ANY KIND, either express or implied. See the License for the specific language
 * governing permissions and limitations under the License.
 */ 
/*
 * Copyright 2020 Adobe. All rights reserved.
 * This file is licensed to you under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License. You may obtain a copy
 * of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under
 * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 * OF ANY KIND, either express or implied. See the License for the specific language
 * governing permissions and limitations under the License.
 */ 
const $cac834c4bc0a51d3$export$c7e742effb1c51e2 = (0, ($parcel$interopDefault($3ThwF$react))).createContext({});
function $cac834c4bc0a51d3$export$21c7ab35b39f78ec() {
    return (0, $3ThwF$react.useContext)($cac834c4bc0a51d3$export$c7e742effb1c51e2);
}
const $cac834c4bc0a51d3$export$e239f1ef5b7afe1a = (0, ($parcel$interopDefault($3ThwF$react))).createContext(undefined);
function $cac834c4bc0a51d3$export$4bbcd0ebecc48abd() {
    return (0, $3ThwF$react.useContext)($cac834c4bc0a51d3$export$e239f1ef5b7afe1a);
}
const $cac834c4bc0a51d3$export$24aad8519b95b41b = (0, ($parcel$interopDefault($3ThwF$react))).createContext({});
function $cac834c4bc0a51d3$export$efa3856fc0e85e7f() {
    return (0, $3ThwF$react.useContext)($cac834c4bc0a51d3$export$24aad8519b95b41b);
}





var $3deee0fd2b46e64f$exports = {};

$parcel$export($3deee0fd2b46e64f$exports, "spectrum-Menu-popover", () => $3deee0fd2b46e64f$export$d959f30786125b, (v) => $3deee0fd2b46e64f$export$d959f30786125b = v);
$parcel$export($3deee0fd2b46e64f$exports, "spectrum-Menu", () => $3deee0fd2b46e64f$export$fae6af6decd50ac5, (v) => $3deee0fd2b46e64f$export$fae6af6decd50ac5 = v);
$parcel$export($3deee0fd2b46e64f$exports, "spectrum-Menu-sectionHeading", () => $3deee0fd2b46e64f$export$c20ea4b35020fbdd, (v) => $3deee0fd2b46e64f$export$c20ea4b35020fbdd = v);
$parcel$export($3deee0fd2b46e64f$exports, "spectrum-Menu-checkmark", () => $3deee0fd2b46e64f$export$8e15a726e4f7e62d, (v) => $3deee0fd2b46e64f$export$8e15a726e4f7e62d = v);
$parcel$export($3deee0fd2b46e64f$exports, "spectrum-Menu-item", () => $3deee0fd2b46e64f$export$45faad328e6c532b, (v) => $3deee0fd2b46e64f$export$45faad328e6c532b = v);
$parcel$export($3deee0fd2b46e64f$exports, "is-selected", () => $3deee0fd2b46e64f$export$1e0fb04f31d3c22a, (v) => $3deee0fd2b46e64f$export$1e0fb04f31d3c22a = v);
$parcel$export($3deee0fd2b46e64f$exports, "spectrum-Icon", () => $3deee0fd2b46e64f$export$d374b04f30360026, (v) => $3deee0fd2b46e64f$export$d374b04f30360026 = v);
$parcel$export($3deee0fd2b46e64f$exports, "spectrum-Menu-itemLabel", () => $3deee0fd2b46e64f$export$1aa6419c5c650567, (v) => $3deee0fd2b46e64f$export$1aa6419c5c650567 = v);
$parcel$export($3deee0fd2b46e64f$exports, "spectrum-Menu-itemIcon", () => $3deee0fd2b46e64f$export$f431001e1bb54633, (v) => $3deee0fd2b46e64f$export$f431001e1bb54633 = v);
$parcel$export($3deee0fd2b46e64f$exports, "spectrum-Menu-itemLabel--wrapping", () => $3deee0fd2b46e64f$export$465092164a8e1d98, (v) => $3deee0fd2b46e64f$export$465092164a8e1d98 = v);
$parcel$export($3deee0fd2b46e64f$exports, "spectrum-Menu-divider", () => $3deee0fd2b46e64f$export$e892026f534efed8, (v) => $3deee0fd2b46e64f$export$e892026f534efed8 = v);
$parcel$export($3deee0fd2b46e64f$exports, "spectrum-Menu-itemGrid", () => $3deee0fd2b46e64f$export$ee5f52784f57f9c8, (v) => $3deee0fd2b46e64f$export$ee5f52784f57f9c8 = v);
$parcel$export($3deee0fd2b46e64f$exports, "is-selectable", () => $3deee0fd2b46e64f$export$fbd22ba224473193, (v) => $3deee0fd2b46e64f$export$fbd22ba224473193 = v);
$parcel$export($3deee0fd2b46e64f$exports, "spectrum-Menu-end", () => $3deee0fd2b46e64f$export$926a1bb74ed65667, (v) => $3deee0fd2b46e64f$export$926a1bb74ed65667 = v);
$parcel$export($3deee0fd2b46e64f$exports, "spectrum-Menu-icon", () => $3deee0fd2b46e64f$export$cbbae63ef9e44c7a, (v) => $3deee0fd2b46e64f$export$cbbae63ef9e44c7a = v);
$parcel$export($3deee0fd2b46e64f$exports, "spectrum-Menu-description", () => $3deee0fd2b46e64f$export$36f1275051e39ead, (v) => $3deee0fd2b46e64f$export$36f1275051e39ead = v);
$parcel$export($3deee0fd2b46e64f$exports, "spectrum-Menu-keyboard", () => $3deee0fd2b46e64f$export$4f9817a7dff6ba0f, (v) => $3deee0fd2b46e64f$export$4f9817a7dff6ba0f = v);
$parcel$export($3deee0fd2b46e64f$exports, "focus-ring", () => $3deee0fd2b46e64f$export$f39a09f249340e2a, (v) => $3deee0fd2b46e64f$export$f39a09f249340e2a = v);
$parcel$export($3deee0fd2b46e64f$exports, "is-focused", () => $3deee0fd2b46e64f$export$e7dc768d35940237, (v) => $3deee0fd2b46e64f$export$e7dc768d35940237 = v);
$parcel$export($3deee0fd2b46e64f$exports, "is-hovered", () => $3deee0fd2b46e64f$export$b8813cd5d7824ce7, (v) => $3deee0fd2b46e64f$export$b8813cd5d7824ce7 = v);
$parcel$export($3deee0fd2b46e64f$exports, "is-highlighted", () => $3deee0fd2b46e64f$export$451c613d0772eb99, (v) => $3deee0fd2b46e64f$export$451c613d0772eb99 = v);
$parcel$export($3deee0fd2b46e64f$exports, "is-open", () => $3deee0fd2b46e64f$export$a9781837241c946d, (v) => $3deee0fd2b46e64f$export$a9781837241c946d = v);
$parcel$export($3deee0fd2b46e64f$exports, "is-active", () => $3deee0fd2b46e64f$export$20fd0f7cd4e6112f, (v) => $3deee0fd2b46e64f$export$20fd0f7cd4e6112f = v);
$parcel$export($3deee0fd2b46e64f$exports, "is-disabled", () => $3deee0fd2b46e64f$export$d35bc1e505d1ebbf, (v) => $3deee0fd2b46e64f$export$d35bc1e505d1ebbf = v);
var $3deee0fd2b46e64f$export$d959f30786125b;
var $3deee0fd2b46e64f$export$fae6af6decd50ac5;
var $3deee0fd2b46e64f$export$c20ea4b35020fbdd;
var $3deee0fd2b46e64f$export$8e15a726e4f7e62d;
var $3deee0fd2b46e64f$export$45faad328e6c532b;
var $3deee0fd2b46e64f$export$1e0fb04f31d3c22a;
var $3deee0fd2b46e64f$export$d374b04f30360026;
var $3deee0fd2b46e64f$export$1aa6419c5c650567;
var $3deee0fd2b46e64f$export$f431001e1bb54633;
var $3deee0fd2b46e64f$export$465092164a8e1d98;
var $3deee0fd2b46e64f$export$e892026f534efed8;
var $3deee0fd2b46e64f$export$ee5f52784f57f9c8;
var $3deee0fd2b46e64f$export$fbd22ba224473193;
var $3deee0fd2b46e64f$export$926a1bb74ed65667;
var $3deee0fd2b46e64f$export$cbbae63ef9e44c7a;
var $3deee0fd2b46e64f$export$36f1275051e39ead;
var $3deee0fd2b46e64f$export$4f9817a7dff6ba0f;
var $3deee0fd2b46e64f$export$f39a09f249340e2a;
var $3deee0fd2b46e64f$export$e7dc768d35940237;
var $3deee0fd2b46e64f$export$b8813cd5d7824ce7;
var $3deee0fd2b46e64f$export$451c613d0772eb99;
var $3deee0fd2b46e64f$export$a9781837241c946d;
var $3deee0fd2b46e64f$export$20fd0f7cd4e6112f;
var $3deee0fd2b46e64f$export$d35bc1e505d1ebbf;
$3deee0fd2b46e64f$export$d959f30786125b = "spectrum-Menu-popover_7ea47c";
$3deee0fd2b46e64f$export$fae6af6decd50ac5 = "spectrum-Menu_7ea47c";
$3deee0fd2b46e64f$export$c20ea4b35020fbdd = "spectrum-Menu-sectionHeading_7ea47c";
$3deee0fd2b46e64f$export$8e15a726e4f7e62d = "spectrum-Menu-checkmark_7ea47c";
$3deee0fd2b46e64f$export$45faad328e6c532b = "spectrum-Menu-item_7ea47c";
$3deee0fd2b46e64f$export$1e0fb04f31d3c22a = "is-selected_7ea47c";
$3deee0fd2b46e64f$export$d374b04f30360026 = "spectrum-Icon_7ea47c";
$3deee0fd2b46e64f$export$1aa6419c5c650567 = "spectrum-Menu-itemLabel_7ea47c";
$3deee0fd2b46e64f$export$f431001e1bb54633 = "spectrum-Menu-itemIcon_7ea47c";
$3deee0fd2b46e64f$export$465092164a8e1d98 = "spectrum-Menu-itemLabel--wrapping_7ea47c";
$3deee0fd2b46e64f$export$e892026f534efed8 = "spectrum-Menu-divider_7ea47c";
$3deee0fd2b46e64f$export$ee5f52784f57f9c8 = "spectrum-Menu-itemGrid_7ea47c";
$3deee0fd2b46e64f$export$fbd22ba224473193 = "is-selectable_7ea47c";
$3deee0fd2b46e64f$export$926a1bb74ed65667 = "spectrum-Menu-end_7ea47c";
$3deee0fd2b46e64f$export$cbbae63ef9e44c7a = "spectrum-Menu-icon_7ea47c";
$3deee0fd2b46e64f$export$36f1275051e39ead = "spectrum-Menu-description_7ea47c";
$3deee0fd2b46e64f$export$4f9817a7dff6ba0f = "spectrum-Menu-keyboard_7ea47c";
$3deee0fd2b46e64f$export$f39a09f249340e2a = "focus-ring_7ea47c";
$3deee0fd2b46e64f$export$e7dc768d35940237 = "is-focused_7ea47c";
$3deee0fd2b46e64f$export$b8813cd5d7824ce7 = "is-hovered_7ea47c";
$3deee0fd2b46e64f$export$451c613d0772eb99 = "is-highlighted_7ea47c";
$3deee0fd2b46e64f$export$a9781837241c946d = "is-open_7ea47c";
$3deee0fd2b46e64f$export$20fd0f7cd4e6112f = "is-active_7ea47c";
$3deee0fd2b46e64f$export$d35bc1e505d1ebbf = "is-disabled_7ea47c";




function $e56311f7204399c7$var$MenuTrigger(props, ref) {
    let triggerRef = (0, $3ThwF$react.useRef)();
    let domRef = (0, $3ThwF$reactspectrumutils.useDOMRef)(ref);
    let menuTriggerRef = domRef || triggerRef;
    let menuRef = (0, $3ThwF$react.useRef)();
    let { children: children , align: align = "start" , shouldFlip: shouldFlip = true , direction: direction = "bottom" , closeOnSelect: closeOnSelect , trigger: trigger = "press"  } = props;
    let [menuTrigger, menu] = (0, ($parcel$interopDefault($3ThwF$react))).Children.toArray(children);
    let state = (0, $3ThwF$reactstatelymenu.useMenuTriggerState)(props);
    let { menuTriggerProps: menuTriggerProps , menuProps: menuProps  } = (0, $3ThwF$reactariamenu.useMenuTrigger)({
        trigger: trigger
    }, state, menuTriggerRef);
    let initialPlacement;
    switch(direction){
        case "left":
        case "right":
        case "start":
        case "end":
            initialPlacement = `${direction} ${align === "end" ? "bottom" : "top"}`;
            break;
        case "bottom":
        case "top":
        default:
            initialPlacement = `${direction} ${align}`;
    }
    let isMobile = (0, $3ThwF$reactspectrumutils.useIsMobileDevice)();
    let menuContext = {
        ...menuProps,
        state: state,
        ref: menuRef,
        onClose: state.close,
        closeOnSelect: closeOnSelect,
        autoFocus: state.focusStrategy || true,
        UNSAFE_style: isMobile ? {
            width: "100%",
            maxHeight: "inherit"
        } : undefined,
        UNSAFE_className: (0, $3ThwF$reactspectrumutils.classNames)((0, (/*@__PURE__*/$parcel$interopDefault($3deee0fd2b46e64f$exports))), {
            "spectrum-Menu-popover": !isMobile
        })
    };
    // On small screen devices, the menu is rendered in a tray, otherwise a popover.
    let overlay;
    if (isMobile) overlay = /*#__PURE__*/ (0, ($parcel$interopDefault($3ThwF$react))).createElement((0, $3ThwF$reactspectrumoverlays.Tray), {
        state: state
    }, menu);
    else overlay = /*#__PURE__*/ (0, ($parcel$interopDefault($3ThwF$react))).createElement((0, $3ThwF$reactspectrumoverlays.Popover), {
        UNSAFE_style: {
            clipPath: "unset"
        },
        state: state,
        triggerRef: menuTriggerRef,
        scrollRef: menuRef,
        placement: initialPlacement,
        hideArrow: true,
        shouldFlip: shouldFlip
    }, menu);
    return /*#__PURE__*/ (0, ($parcel$interopDefault($3ThwF$react))).createElement((0, $3ThwF$react.Fragment), null, /*#__PURE__*/ (0, ($parcel$interopDefault($3ThwF$react))).createElement((0, $3ThwF$reactspectrumutils.SlotProvider), {
        slots: {
            actionButton: {
                holdAffordance: trigger === "longPress"
            }
        }
    }, /*#__PURE__*/ (0, ($parcel$interopDefault($3ThwF$react))).createElement((0, $3ThwF$reactariainteractions.PressResponder), {
        ...menuTriggerProps,
        ref: menuTriggerRef,
        isPressed: state.isOpen
    }, menuTrigger)), /*#__PURE__*/ (0, ($parcel$interopDefault($3ThwF$react))).createElement((0, $cac834c4bc0a51d3$export$c7e742effb1c51e2).Provider, {
        value: menuContext
    }, overlay));
}
/**
 * The MenuTrigger serves as a wrapper around a Menu and its associated trigger,
 * linking the Menu's open state with the trigger's press state.
 */ let $e56311f7204399c7$export$27d2ad3c5815583e = /*#__PURE__*/ (0, $3ThwF$react.forwardRef)($e56311f7204399c7$var$MenuTrigger);


/*
 * Copyright 2020 Adobe. All rights reserved.
 * This file is licensed to you under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License. You may obtain a copy
 * of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under
 * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 * OF ANY KIND, either express or implied. See the License for the specific language
 * governing permissions and limitations under the License.
 */ 


/*
 * Copyright 2020 Adobe. All rights reserved.
 * This file is licensed to you under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License. You may obtain a copy
 * of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under
 * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 * OF ANY KIND, either express or implied. See the License for the specific language
 * governing permissions and limitations under the License.
 */ 





var $2f9ceea24fba5443$exports = {};
var $517603d8c7b8e5e5$exports = {};
$517603d8c7b8e5e5$exports = {
    "moreActions": `المزيد من الإجراءات`,
    "unavailable": `قُم بالتوسيع للحصول على التفاصيل`
};


var $da4736ecb2b2fc3d$exports = {};
$da4736ecb2b2fc3d$exports = {
    "moreActions": `Повече действия`,
    "unavailable": `Разширете за подробности`
};


var $474902e64d15239d$exports = {};
$474902e64d15239d$exports = {
    "moreActions": `Další akce`,
    "unavailable": `Podrobnosti zobrazíte rozbalením`
};


var $ab827474128c0a44$exports = {};
$ab827474128c0a44$exports = {
    "moreActions": `Flere handlinger`,
    "unavailable": `Udvid for detaljer`
};


var $6decd364a8a60c6f$exports = {};
$6decd364a8a60c6f$exports = {
    "moreActions": `Mehr Aktionen`,
    "unavailable": `Für Details erweitern`
};


var $a79ffc3f42b40ea0$exports = {};
$a79ffc3f42b40ea0$exports = {
    "moreActions": `Περισσότερες ενέργειες`,
    "unavailable": `Ανάπτυξη για λεπτομέρειες`
};


var $2b0d2ea968c3f427$exports = {};
$2b0d2ea968c3f427$exports = {
    "moreActions": `More actions`,
    "unavailable": `Expand for details`
};


var $3655ccfea2cb795f$exports = {};
$3655ccfea2cb795f$exports = {
    "moreActions": `Más acciones`,
    "unavailable": `Ampliar para más detalles`
};


var $fdd5fdd03c65691f$exports = {};
$fdd5fdd03c65691f$exports = {
    "moreActions": `Veel toiminguid`,
    "unavailable": `Laiendage üksikasjade kuvamiseks`
};


var $5b53a5d82dacf660$exports = {};
$5b53a5d82dacf660$exports = {
    "moreActions": `Lisää toimintoja`,
    "unavailable": `Laajenna lisätietoja`
};


var $53cf7303c3810c0d$exports = {};
$53cf7303c3810c0d$exports = {
    "moreActions": `Autres actions`,
    "unavailable": `Agrandir pour plus de détails`
};


var $1e6d66e0b4792817$exports = {};
$1e6d66e0b4792817$exports = {
    "moreActions": `פעולות נוספות`,
    "unavailable": `הרחב לפרטים`
};


var $eb204bb933ebd45f$exports = {};
$eb204bb933ebd45f$exports = {
    "moreActions": `Dodatne radnje`,
    "unavailable": `Proširite za detalje`
};


var $3b770de73ab860bc$exports = {};
$3b770de73ab860bc$exports = {
    "moreActions": `További lehetőségek`,
    "unavailable": `A részletekért bontsa ki`
};


var $010ff727ed3102ae$exports = {};
$010ff727ed3102ae$exports = {
    "moreActions": `Altre azioni`,
    "unavailable": `Espandi per i dettagli`
};


var $dead61005b56a753$exports = {};
$dead61005b56a753$exports = {
    "moreActions": `その他のアクション`,
    "unavailable": `展開して詳細を見る`
};


var $059a8cc6da15c9c5$exports = {};
$059a8cc6da15c9c5$exports = {
    "moreActions": `기타 액션`,
    "unavailable": `자세히 보려면 펼치기`
};


var $115228ca3cb07b54$exports = {};
$115228ca3cb07b54$exports = {
    "moreActions": `Daugiau veiksmų`,
    "unavailable": `Norėdami gauti daugiau informacijos, išskleiskite`
};


var $1b70a2b594f0946f$exports = {};
$1b70a2b594f0946f$exports = {
    "moreActions": `Citas darbības`,
    "unavailable": `Izvērst, lai iegūtu sīkāku informāciju`
};


var $10afb7333b6cd21a$exports = {};
$10afb7333b6cd21a$exports = {
    "moreActions": `Flere handlinger`,
    "unavailable": `Utvid for detaljer`
};


var $6fb3df0b34cc117d$exports = {};
$6fb3df0b34cc117d$exports = {
    "moreActions": `Meer handelingen`,
    "unavailable": `Uitvouwen voor meer informatie`
};


var $f6cf802e78a19c98$exports = {};
$f6cf802e78a19c98$exports = {
    "moreActions": `Więcej akcji`,
    "unavailable": `Rozwiń, aby zobaczyć szczegóły`
};


var $7401b34fc83fec2d$exports = {};
$7401b34fc83fec2d$exports = {
    "moreActions": `Mais ações`,
    "unavailable": `Expandir para ver detalhes`
};


var $812f012e4938f38d$exports = {};
$812f012e4938f38d$exports = {
    "moreActions": `Mais ações`,
    "unavailable": `Expandir para obter mais pormenores`
};


var $8c983e7e2aa14f7a$exports = {};
$8c983e7e2aa14f7a$exports = {
    "moreActions": `Mai multe acțiuni`,
    "unavailable": `Extindeți pentru detalii`
};


var $5eddd4454bdacc3e$exports = {};
$5eddd4454bdacc3e$exports = {
    "moreActions": `Дополнительные действия`,
    "unavailable": `Развернуть для подробностей`
};


var $3dc143fbc7df8c0b$exports = {};
$3dc143fbc7df8c0b$exports = {
    "moreActions": `Ďalšie akcie`,
    "unavailable": `Rozbaľte a zobrazte podrobnosti`
};


var $089b9b9eacba8462$exports = {};
$089b9b9eacba8462$exports = {
    "moreActions": `Več možnosti`,
    "unavailable": `Razširite za podrobnosti`
};


var $ae817dc85563e7c9$exports = {};
$ae817dc85563e7c9$exports = {
    "moreActions": `Dodatne radnje`,
    "unavailable": `Expand for details`
};


var $39f858f9383858ce$exports = {};
$39f858f9383858ce$exports = {
    "moreActions": `Fler åtgärder`,
    "unavailable": `Expandera för information`
};


var $fb520ca0abee09d8$exports = {};
$fb520ca0abee09d8$exports = {
    "moreActions": `Daha fazla eylem`,
    "unavailable": `Ayrıntıları görmek için genişletin`
};


var $7950b4249316e2c0$exports = {};
$7950b4249316e2c0$exports = {
    "moreActions": `Більше дій`,
    "unavailable": `Розгорніть для докладнішої інформації`
};


var $86c8e6b13fa9c9ad$exports = {};
$86c8e6b13fa9c9ad$exports = {
    "moreActions": `更多操作`,
    "unavailable": `展开以查看详细信息`
};


var $f3cd1d6d75907b41$exports = {};
$f3cd1d6d75907b41$exports = {
    "moreActions": `更多動作`,
    "unavailable": `展開以取得詳細資料`
};


$2f9ceea24fba5443$exports = {
    "ar-AE": $517603d8c7b8e5e5$exports,
    "bg-BG": $da4736ecb2b2fc3d$exports,
    "cs-CZ": $474902e64d15239d$exports,
    "da-DK": $ab827474128c0a44$exports,
    "de-DE": $6decd364a8a60c6f$exports,
    "el-GR": $a79ffc3f42b40ea0$exports,
    "en-US": $2b0d2ea968c3f427$exports,
    "es-ES": $3655ccfea2cb795f$exports,
    "et-EE": $fdd5fdd03c65691f$exports,
    "fi-FI": $5b53a5d82dacf660$exports,
    "fr-FR": $53cf7303c3810c0d$exports,
    "he-IL": $1e6d66e0b4792817$exports,
    "hr-HR": $eb204bb933ebd45f$exports,
    "hu-HU": $3b770de73ab860bc$exports,
    "it-IT": $010ff727ed3102ae$exports,
    "ja-JP": $dead61005b56a753$exports,
    "ko-KR": $059a8cc6da15c9c5$exports,
    "lt-LT": $115228ca3cb07b54$exports,
    "lv-LV": $1b70a2b594f0946f$exports,
    "nb-NO": $10afb7333b6cd21a$exports,
    "nl-NL": $6fb3df0b34cc117d$exports,
    "pl-PL": $f6cf802e78a19c98$exports,
    "pt-BR": $7401b34fc83fec2d$exports,
    "pt-PT": $812f012e4938f38d$exports,
    "ro-RO": $8c983e7e2aa14f7a$exports,
    "ru-RU": $5eddd4454bdacc3e$exports,
    "sk-SK": $3dc143fbc7df8c0b$exports,
    "sl-SI": $089b9b9eacba8462$exports,
    "sr-SP": $ae817dc85563e7c9$exports,
    "sv-SE": $39f858f9383858ce$exports,
    "tr-TR": $fb520ca0abee09d8$exports,
    "uk-UA": $7950b4249316e2c0$exports,
    "zh-CN": $86c8e6b13fa9c9ad$exports,
    "zh-TW": $f3cd1d6d75907b41$exports
};








function $1d718563708db5b5$export$2ce376c2cc3355c8(props) {
    let { item: item , state: state , isVirtualized: isVirtualized , onAction: onAction  } = props;
    let stringFormatter = (0, $3ThwF$reactariai18n.useLocalizedStringFormatter)((0, (/*@__PURE__*/$parcel$interopDefault($2f9ceea24fba5443$exports))));
    let menuDialogContext = (0, $cac834c4bc0a51d3$export$4bbcd0ebecc48abd)();
    let { triggerRef: triggerRef  } = menuDialogContext || {};
    let isMenuDialogTrigger = !!menuDialogContext;
    let isUnavailable = false;
    if (isMenuDialogTrigger) isUnavailable = menuDialogContext.isUnavailable;
    let domProps = (0, $3ThwF$reactariautils.filterDOMProps)(item.props);
    let { onClose: onClose , closeOnSelect: closeOnSelect  } = (0, $cac834c4bc0a51d3$export$21c7ab35b39f78ec)();
    let { rendered: rendered , key: key  } = item;
    let isSelected = state.selectionManager.isSelected(key);
    let isDisabled = state.disabledKeys.has(key);
    let itemref = (0, $3ThwF$react.useRef)(null);
    let ref = (0, $3ThwF$reactariautils.useObjectRef)((0, $3ThwF$react.useMemo)(()=>(0, $3ThwF$reactariautils.mergeRefs)(itemref, triggerRef), [
        itemref,
        triggerRef
    ]));
    let { menuItemProps: menuItemProps , labelProps: labelProps , descriptionProps: descriptionProps , keyboardShortcutProps: keyboardShortcutProps  } = (0, $3ThwF$reactariamenu.useMenuItem)({
        isSelected: isSelected,
        isDisabled: isDisabled,
        "aria-label": item["aria-label"],
        key: key,
        onClose: onClose,
        closeOnSelect: closeOnSelect,
        isVirtualized: isVirtualized,
        onAction: onAction,
        "aria-haspopup": isMenuDialogTrigger && isUnavailable ? "dialog" : undefined
    }, state, ref);
    let endId = (0, $3ThwF$reactariautils.useSlotId)();
    let endProps = {};
    if (endId) {
        endProps.id = endId;
        menuItemProps["aria-describedby"] = [
            menuItemProps["aria-describedby"],
            endId
        ].filter(Boolean).join(" ");
    }
    if (isUnavailable) menuItemProps["aria-disabled"] = "true";
    let contents = typeof rendered === "string" ? /*#__PURE__*/ (0, ($parcel$interopDefault($3ThwF$react))).createElement((0, $3ThwF$reactspectrumtext.Text), null, rendered) : rendered;
    return /*#__PURE__*/ (0, ($parcel$interopDefault($3ThwF$react))).createElement((0, $3ThwF$reactariafocus.FocusRing), {
        focusRingClass: (0, $3ThwF$reactspectrumutils.classNames)((0, (/*@__PURE__*/$parcel$interopDefault($3deee0fd2b46e64f$exports))), "focus-ring")
    }, /*#__PURE__*/ (0, ($parcel$interopDefault($3ThwF$react))).createElement("li", {
        ...(0, $3ThwF$reactariautils.mergeProps)(menuItemProps, domProps),
        ref: ref,
        className: (0, $3ThwF$reactspectrumutils.classNames)((0, (/*@__PURE__*/$parcel$interopDefault($3deee0fd2b46e64f$exports))), "spectrum-Menu-item", {
            "is-disabled": isDisabled,
            "is-selected": isSelected,
            "is-selectable": state.selectionManager.selectionMode !== "none",
            "is-open": state.expandedKeys.has(key)
        })
    }, /*#__PURE__*/ (0, ($parcel$interopDefault($3ThwF$react))).createElement((0, $3ThwF$reactspectrumlayout.Grid), {
        UNSAFE_className: (0, $3ThwF$reactspectrumutils.classNames)((0, (/*@__PURE__*/$parcel$interopDefault($3deee0fd2b46e64f$exports))), "spectrum-Menu-itemGrid")
    }, /*#__PURE__*/ (0, ($parcel$interopDefault($3ThwF$react))).createElement((0, $3ThwF$reactspectrumutils.ClearSlots), null, /*#__PURE__*/ (0, ($parcel$interopDefault($3ThwF$react))).createElement((0, $3ThwF$reactspectrumutils.SlotProvider), {
        slots: {
            text: {
                UNSAFE_className: (0, (/*@__PURE__*/$parcel$interopDefault($3deee0fd2b46e64f$exports)))["spectrum-Menu-itemLabel"],
                ...labelProps
            },
            end: {
                UNSAFE_className: (0, (/*@__PURE__*/$parcel$interopDefault($3deee0fd2b46e64f$exports)))["spectrum-Menu-end"],
                ...endProps
            },
            icon: {
                UNSAFE_className: (0, (/*@__PURE__*/$parcel$interopDefault($3deee0fd2b46e64f$exports)))["spectrum-Menu-icon"],
                size: "S"
            },
            description: {
                UNSAFE_className: (0, (/*@__PURE__*/$parcel$interopDefault($3deee0fd2b46e64f$exports)))["spectrum-Menu-description"],
                ...descriptionProps
            },
            keyboard: {
                UNSAFE_className: (0, (/*@__PURE__*/$parcel$interopDefault($3deee0fd2b46e64f$exports)))["spectrum-Menu-keyboard"],
                ...keyboardShortcutProps
            }
        }
    }, contents, isSelected && /*#__PURE__*/ (0, ($parcel$interopDefault($3ThwF$react))).createElement((0, ($parcel$interopDefault($3ThwF$spectrumiconsuiCheckmarkMedium))), {
        slot: "checkmark",
        UNSAFE_className: (0, $3ThwF$reactspectrumutils.classNames)((0, (/*@__PURE__*/$parcel$interopDefault($3deee0fd2b46e64f$exports))), "spectrum-Menu-checkmark")
    }), isUnavailable && /*#__PURE__*/ (0, ($parcel$interopDefault($3ThwF$react))).createElement((0, ($parcel$interopDefault($3ThwF$spectrumiconsworkflowInfoOutline))), {
        slot: "end",
        size: "XS",
        alignSelf: "center",
        "aria-label": stringFormatter.format("unavailable")
    }))))));
}


/*
 * Copyright 2020 Adobe. All rights reserved.
 * This file is licensed to you under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License. You may obtain a copy
 * of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under
 * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 * OF ANY KIND, either express or implied. See the License for the specific language
 * governing permissions and limitations under the License.
 */ 






function $5793ff412227d2fc$export$4b1545b4f2016d26(props) {
    let { item: item , state: state , onAction: onAction  } = props;
    let { itemProps: itemProps , headingProps: headingProps , groupProps: groupProps  } = (0, $3ThwF$reactariamenu.useMenuSection)({
        heading: item.rendered,
        "aria-label": item["aria-label"]
    });
    let { separatorProps: separatorProps  } = (0, $3ThwF$reactariaseparator.useSeparator)({
        elementType: "li"
    });
    return /*#__PURE__*/ (0, ($parcel$interopDefault($3ThwF$react))).createElement((0, $3ThwF$react.Fragment), null, item.key !== state.collection.getFirstKey() && /*#__PURE__*/ (0, ($parcel$interopDefault($3ThwF$react))).createElement("li", {
        ...separatorProps,
        className: (0, $3ThwF$reactspectrumutils.classNames)((0, (/*@__PURE__*/$parcel$interopDefault($3deee0fd2b46e64f$exports))), "spectrum-Menu-divider")
    }), /*#__PURE__*/ (0, ($parcel$interopDefault($3ThwF$react))).createElement("li", itemProps, item.rendered && /*#__PURE__*/ (0, ($parcel$interopDefault($3ThwF$react))).createElement("span", {
        ...headingProps,
        className: (0, $3ThwF$reactspectrumutils.classNames)((0, (/*@__PURE__*/$parcel$interopDefault($3deee0fd2b46e64f$exports))), "spectrum-Menu-sectionHeading")
    }, item.rendered), /*#__PURE__*/ (0, ($parcel$interopDefault($3ThwF$react))).createElement("ul", {
        ...groupProps,
        className: (0, $3ThwF$reactspectrumutils.classNames)((0, (/*@__PURE__*/$parcel$interopDefault($3deee0fd2b46e64f$exports))), "spectrum-Menu")
    }, [
        ...(0, $3ThwF$reactstatelycollections.getChildNodes)(item, state.collection)
    ].map((node)=>{
        let item = /*#__PURE__*/ (0, ($parcel$interopDefault($3ThwF$react))).createElement((0, $1d718563708db5b5$export$2ce376c2cc3355c8), {
            key: node.key,
            item: node,
            state: state,
            onAction: onAction
        });
        if (node.wrapper) item = node.wrapper(item);
        return item;
    }))));
}







function $63a7dff9cbe2d046$var$Menu(props, ref) {
    let contextProps = (0, $3ThwF$react.useContext)((0, $cac834c4bc0a51d3$export$c7e742effb1c51e2));
    let completeProps = {
        ...(0, $3ThwF$reactariautils.mergeProps)(contextProps, props)
    };
    let domRef = (0, $3ThwF$reactspectrumutils.useDOMRef)(ref);
    let scopedRef = (0, $3ThwF$react.useRef)(null);
    let state = (0, $3ThwF$reactstatelytree.useTreeState)(completeProps);
    let { menuProps: menuProps  } = (0, $3ThwF$reactariamenu.useMenu)(completeProps, state, domRef);
    let { styleProps: styleProps  } = (0, $3ThwF$reactspectrumutils.useStyleProps)(completeProps);
    (0, $3ThwF$reactariautils.useSyncRef)(contextProps, domRef);
    return /*#__PURE__*/ (0, ($parcel$interopDefault($3ThwF$react))).createElement((0, $cac834c4bc0a51d3$export$24aad8519b95b41b).Provider, {
        value: {
            state: state,
            container: scopedRef,
            menu: domRef
        }
    }, /*#__PURE__*/ (0, ($parcel$interopDefault($3ThwF$react))).createElement((0, $3ThwF$reactariafocus.FocusScope), {
        contain: state.expandedKeys.size > 0
    }, /*#__PURE__*/ (0, ($parcel$interopDefault($3ThwF$react))).createElement("ul", {
        ...menuProps,
        ...styleProps,
        ref: domRef,
        className: (0, $3ThwF$reactspectrumutils.classNames)((0, (/*@__PURE__*/$parcel$interopDefault($3deee0fd2b46e64f$exports))), "spectrum-Menu", styleProps.className)
    }, [
        ...state.collection
    ].map((item)=>{
        if (item.type === "section") return /*#__PURE__*/ (0, ($parcel$interopDefault($3ThwF$react))).createElement((0, $5793ff412227d2fc$export$4b1545b4f2016d26), {
            key: item.key,
            item: item,
            state: state,
            onAction: completeProps.onAction
        });
        let menuItem = /*#__PURE__*/ (0, ($parcel$interopDefault($3ThwF$react))).createElement((0, $1d718563708db5b5$export$2ce376c2cc3355c8), {
            key: item.key,
            item: item,
            state: state,
            onAction: completeProps.onAction
        });
        if (item.wrapper) menuItem = item.wrapper(menuItem);
        return menuItem;
    })), /*#__PURE__*/ (0, ($parcel$interopDefault($3ThwF$react))).createElement("div", {
        ref: scopedRef
    })));
}
/**
 * Menus display a list of actions or options that a user can choose.
 */ // forwardRef doesn't support generic parameters, so cast the result to the correct type
// https://stackoverflow.com/questions/58469229/react-with-typescript-generics-while-using-react-forwardref
const $63a7dff9cbe2d046$export$d9b273488cd8ce6f = /*#__PURE__*/ (0, ($parcel$interopDefault($3ThwF$react))).forwardRef($63a7dff9cbe2d046$var$Menu);


/*
 * Copyright 2021 Adobe. All rights reserved.
 * This file is licensed to you under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License. You may obtain a copy
 * of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under
 * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 * OF ANY KIND, either express or implied. See the License for the specific language
 * governing permissions and limitations under the License.
 */ 








function $53a2f2636c97a561$var$ActionMenu(props, ref) {
    props = (0, $3ThwF$reactspectrumutils.useSlotProps)(props, "actionMenu");
    let stringFormatter = (0, $3ThwF$reactariai18n.useLocalizedStringFormatter)((0, (/*@__PURE__*/$parcel$interopDefault($2f9ceea24fba5443$exports))));
    let buttonProps = (0, $3ThwF$reactariautils.filterDOMProps)(props, {
        labelable: true
    });
    if (buttonProps["aria-label"] === undefined) buttonProps["aria-label"] = stringFormatter.format("moreActions");
    return /*#__PURE__*/ (0, ($parcel$interopDefault($3ThwF$react))).createElement((0, $e56311f7204399c7$export$27d2ad3c5815583e), {
        isOpen: props.isOpen,
        defaultOpen: props.defaultOpen,
        onOpenChange: props.onOpenChange,
        align: props.align,
        direction: props.direction,
        shouldFlip: props.shouldFlip
    }, /*#__PURE__*/ (0, ($parcel$interopDefault($3ThwF$react))).createElement((0, $3ThwF$reactspectrumbutton.ActionButton), {
        ref: ref,
        ...props,
        ...buttonProps
    }, /*#__PURE__*/ (0, ($parcel$interopDefault($3ThwF$react))).createElement((0, ($parcel$interopDefault($3ThwF$spectrumiconsworkflowMore))), null)), /*#__PURE__*/ (0, ($parcel$interopDefault($3ThwF$react))).createElement((0, $63a7dff9cbe2d046$export$d9b273488cd8ce6f), {
        children: props.children,
        items: props.items,
        disabledKeys: props.disabledKeys,
        onAction: props.onAction
    }));
}
/**
 * ActionMenu combines an ActionButton with a Menu for simple "more actions" use cases.
 */ const $53a2f2636c97a561$export$ed57a210b9d13bb6 = /*#__PURE__*/ (0, $3ThwF$react.forwardRef)($53a2f2636c97a561$var$ActionMenu);


/*
 * Copyright 2023 Adobe. All rights reserved.
 * This file is licensed to you under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License. You may obtain a copy
 * of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under
 * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 * OF ANY KIND, either express or implied. See the License for the specific language
 * governing permissions and limitations under the License.
 */ 

var $a10e565ec42bcf96$exports = {};

$parcel$export($a10e565ec42bcf96$exports, "react-spectrum-ContextualHelp-button", () => $a10e565ec42bcf96$export$20d9583c9f1529c1, (v) => $a10e565ec42bcf96$export$20d9583c9f1529c1 = v);
$parcel$export($a10e565ec42bcf96$exports, "react-spectrum-ContextualHelp-dialog", () => $a10e565ec42bcf96$export$ab2d873674d20d28, (v) => $a10e565ec42bcf96$export$ab2d873674d20d28 = v);
$parcel$export($a10e565ec42bcf96$exports, "react-spectrum-ContextualHelp-content", () => $a10e565ec42bcf96$export$a1f805890cf58c4, (v) => $a10e565ec42bcf96$export$a1f805890cf58c4 = v);
$parcel$export($a10e565ec42bcf96$exports, "react-spectrum-ContextualHelp-footer", () => $a10e565ec42bcf96$export$d901bc3fa82b87a6, (v) => $a10e565ec42bcf96$export$d901bc3fa82b87a6 = v);
var $a10e565ec42bcf96$export$20d9583c9f1529c1;
var $a10e565ec42bcf96$export$ab2d873674d20d28;
var $a10e565ec42bcf96$export$a1f805890cf58c4;
var $a10e565ec42bcf96$export$d901bc3fa82b87a6;
$a10e565ec42bcf96$export$20d9583c9f1529c1 = "react-spectrum-ContextualHelp-button_0cd31c";
$a10e565ec42bcf96$export$ab2d873674d20d28 = "react-spectrum-ContextualHelp-dialog_0cd31c";
$a10e565ec42bcf96$export$a1f805890cf58c4 = "react-spectrum-ContextualHelp-content_0cd31c";
$a10e565ec42bcf96$export$d901bc3fa82b87a6 = "react-spectrum-ContextualHelp-footer_0cd31c";






function $db380858d398b658$var$ContextualHelpTrigger(props) {
    let { isUnavailable: isUnavailable  } = props;
    let triggerRef = (0, $3ThwF$react.useRef)(null);
    let popoverRef = (0, $3ThwF$react.useRef)(null);
    let { state: menuState , container: container , menu: menu  } = (0, $cac834c4bc0a51d3$export$efa3856fc0e85e7f)();
    let state = (0, $3ThwF$reactstatelyoverlays.useOverlayTriggerState)({
        isOpen: menuState.expandedKeys.has(props.targetKey),
        onOpenChange: (val)=>{
            if (!val) {
                if (menuState.expandedKeys.has(props.targetKey)) menuState.toggleKey(props.targetKey);
            }
        }
    });
    let slots = {};
    if (isUnavailable) slots = {
        dialog: {
            UNSAFE_className: (0, $3ThwF$reactspectrumutils.classNames)((0, (/*@__PURE__*/$parcel$interopDefault($a10e565ec42bcf96$exports))), "react-spectrum-ContextualHelp-dialog")
        },
        content: {
            UNSAFE_className: (0, (/*@__PURE__*/$parcel$interopDefault($a10e565ec42bcf96$exports)))["react-spectrum-ContextualHelp-content"]
        },
        footer: {
            UNSAFE_className: (0, (/*@__PURE__*/$parcel$interopDefault($a10e565ec42bcf96$exports)))["react-spectrum-ContextualHelp-footer"]
        }
    };
    let [trigger] = (0, ($parcel$interopDefault($3ThwF$react))).Children.toArray(props.children);
    let [, content] = props.children;
    let isMobile = (0, $3ThwF$reactspectrumutils.useIsMobileDevice)();
    let onExit = ()=>{
        // if focus was already moved back to a menu item, don't need to do anything
        if (!menu.current.contains(document.activeElement)) // need to return focus to the trigger because hitting Esc causes focus to go to the subdialog, which is then unmounted
        // this leads to blur never being fired nor focus on the body
        triggerRef.current.focus();
    };
    let onBlurWithin = (e)=>{
        var _popoverRef_current;
        if (e.relatedTarget && popoverRef.current && !((_popoverRef_current = popoverRef.current) === null || _popoverRef_current === void 0 ? void 0 : _popoverRef_current.UNSAFE_getDOMNode().contains(e.relatedTarget))) {
            if (menuState.expandedKeys.has(props.targetKey)) menuState.toggleKey(props.targetKey);
        }
    };
    return /*#__PURE__*/ (0, ($parcel$interopDefault($3ThwF$react))).createElement((0, ($parcel$interopDefault($3ThwF$react))).Fragment, null, /*#__PURE__*/ (0, ($parcel$interopDefault($3ThwF$react))).createElement((0, $cac834c4bc0a51d3$export$e239f1ef5b7afe1a).Provider, {
        value: {
            isUnavailable: isUnavailable,
            triggerRef: triggerRef
        }
    }, trigger), /*#__PURE__*/ (0, ($parcel$interopDefault($3ThwF$react))).createElement((0, $3ThwF$reactspectrumutils.SlotProvider), {
        slots: slots
    }, isMobile ? /*#__PURE__*/ (0, ($parcel$interopDefault($3ThwF$react))).createElement((0, $3ThwF$reactspectrumoverlays.Modal), {
        state: state,
        isDismissable: true
    }, /*#__PURE__*/ (0, ($parcel$interopDefault($3ThwF$react))).createElement((0, $3ThwF$reactariaoverlays.DismissButton), {
        onDismiss: state.close
    }), content, /*#__PURE__*/ (0, ($parcel$interopDefault($3ThwF$react))).createElement((0, $3ThwF$reactariaoverlays.DismissButton), {
        onDismiss: state.close
    })) : /*#__PURE__*/ (0, ($parcel$interopDefault($3ThwF$react))).createElement((0, $3ThwF$reactspectrumoverlays.Popover), {
        onExit: onExit,
        onBlurWithin: onBlurWithin,
        container: container.current,
        state: state,
        ref: popoverRef,
        triggerRef: triggerRef,
        placement: "end top",
        offset: -10,
        hideArrow: true,
        isNonModal: true,
        enableBothDismissButtons: true,
        disableFocusManagement: true
    }, content)));
}
$db380858d398b658$var$ContextualHelpTrigger.getCollectionNode = function* getCollectionNode(props) {
    let [trigger] = (0, ($parcel$interopDefault($3ThwF$react))).Children.toArray(props.children);
    let [, content] = props.children;
    yield {
        element: /*#__PURE__*/ (0, ($parcel$interopDefault($3ThwF$react))).cloneElement(trigger, {
            ...trigger.props,
            hasChildItems: true
        }),
        wrapper: (element)=>/*#__PURE__*/ (0, ($parcel$interopDefault($3ThwF$react))).createElement($db380858d398b658$var$ContextualHelpTrigger, {
                key: element.key,
                targetKey: element.key,
                ...props
            }, element, content)
    };
};
let $db380858d398b658$export$5413b169fff83e61 = $db380858d398b658$var$ContextualHelpTrigger;
